diff -ruN nethack-3.4.0.orig/include/extern.h nethack-3.4.0/include/extern.h
--- nethack-3.4.0.orig/include/extern.h	Sat Apr 13 17:17:13 2002
+++ nethack-3.4.0/include/extern.h	Sat Apr 13 17:30:36 2002
@@ -677,6 +677,7 @@
 E char *FDECL(in_rooms, (XCHAR_P,XCHAR_P,int));
 E void FDECL(check_special_room, (BOOLEAN_P));
 E int NDECL(dopickup);
+E int NDECL(dowashhands);
 E void NDECL(lookaround);
 E int NDECL(monster_nearby);
 E void FDECL(nomul, (int));
diff -ruN nethack-3.4.0.orig/src/cmd.c nethack-3.4.0/src/cmd.c
--- nethack-3.4.0.orig/src/cmd.c	Sat Apr 13 17:17:13 2002
+++ nethack-3.4.0/src/cmd.c	Sat Apr 13 17:31:50 2002
@@ -97,6 +97,7 @@
 extern int NDECL(dowieldquiver); /**/
 extern int NDECL(dozap); /**/
 extern int NDECL(doorganize); /**/
+extern int NDECL(dowashhands); /**/
 #ifdef STICKY_OBJECTS
 extern int NDECL(dosticky); /**/
 #endif /* STICKY_OBJECTS */
@@ -1800,6 +1801,7 @@
 	{"untrap", "untrap something", dountrap, FALSE},
 	{"version", "list compile time options for this version of NetHack",
 		doextversion, TRUE},
+        {"wash", "wash your hands", dowashhands, FALSE},
 	{"wipe", "wipe off your face", dowipe, FALSE},
 	{"?", "get this list of extended commands", doextlist, TRUE},
 #if defined(WIZARD)
diff -ruN nethack-3.4.0.orig/src/do_wear.c nethack-3.4.0/src/do_wear.c
--- nethack-3.4.0.orig/src/do_wear.c	Thu Apr 11 17:58:28 2002
+++ nethack-3.4.0/src/do_wear.c	Sat Apr 13 17:30:36 2002
@@ -1599,6 +1599,11 @@
 		if (otmp->otyp != LOADSTONE || !otmp->cursed)
 			dropx(otmp);
 	}
+        if (Glib && Underwater) {
+		Glib = 0;
+		Your("%s are much cleaner now.",
+			makeplural(body_part(HAND)));
+	}
 }
 
 struct obj *
diff -ruN nethack-3.4.0.orig/src/hack.c nethack-3.4.0/src/hack.c
--- nethack-3.4.0.orig/src/hack.c	Sat Apr 13 17:19:02 2002
+++ nethack-3.4.0/src/hack.c	Sat Apr 13 17:30:36 2002
@@ -443,6 +443,66 @@
 	newsym(ox, oy);
 }
 
+int
+dowashhands()
+{
+ register struct rm *lev = &levl[u.ux][u.uy];
+ 
+ if (check_capacity((char *)0)) return (0);
+ if (nohands(youmonst.data)) {
+    You("have no hands.");
+    return 0;
+ }
+ if (Underwater) {
+   pline("What, are you keeping your %s %s?",
+      makeplural(body_part(HAND)),
+      (Is_waterlevel(&u.uz) ? "out of the water" : "above the water level"));
+   return (0);
+ }
+ if (IS_POOL(lev->typ) || IS_FOUNTAIN(lev->typ)
+#ifdef SINKS     
+     || IS_SINK(lev->typ)
+#endif
+     ) {
+   if (Levitation && !Lev_at_will) {
+      You_cant("reach the water.");
+      return (0);
+   }
+   if (uwep && uwep->cursed || uswapwep && uswapwep->cursed) {
+      You_cant("do a proper scrubbing without both hands free.");
+      return (0);
+   }
+   if (uarmg) {
+      /* Glib state is not maintained separately for gloves and hands */
+      You("clean your gloves in the water.");
+   } else 
+   if (noncorporeal(youmonst.data) || unsolid(youmonst.data)) {
+      pline_The("water flows through your %s.",makeplural(body_part(HAND)));
+   } else {
+      You("%s your %s%s.",
+	  (Role_if(PM_HEALER) ? "scrub" : "wash"),
+	  makeplural(body_part(HAND)),
+	  (Glib ? " clean" : ", but they don't get any cleaner"));
+#ifdef TOURIST
+      /* Tourist is the nearest thing to a hitchhiker? */
+      if (!carrying(TOWEL) && Role_if(PM_TOURIST))
+         pline("Do you know where your towel is?");
+#endif
+   }
+   Glib = 0;
+   /* Monsters with flames don't have hands, otherwise could deal 
+    * few points of damage: "The water extinguishes [some of] your flames!"
+    * (What about salamanders?)
+    */
+   return (1);
+ } else {
+       You_cant("%s any water %s.",
+	     (Blind ? "feel" : "see"),
+ 	     (Is_waterlevel(&u.uz) ? "close enough" : "here"));
+   return (0);
+ }
+}
+
 #ifdef SINKS
 static NEARDATA const char fell_on_sink[] = "fell onto a sink";
 
