diff -ruN nethack-3.4.0./dat/dungeon.def nethack-3.4.0/dat/dungeon.def
--- nethack-3.4.0./dat/dungeon.def	Thu Apr 11 17:58:27 2002
+++ nethack-3.4.0/dat/dungeon.def	Sun Jun 16 21:31:09 2002
@@ -17,6 +17,7 @@
 DUNGEON:	"The Dungeons of Doom" "D" (25, 5)
 ALIGNMENT:	unaligned
 BRANCH:		"The Gnomish Mines" @ (2, 3)
+RNDLEVEL:	"rnd1" "5" @ (3, 3) 11
 %REINCARNATION LEVEL:		"rogue" "R" @ (15, 4)
 LEVEL:		"oracle" "O" @ (5, 5)
 LEVALIGN:	neutral
@@ -70,9 +71,9 @@
 DUNGEON:	"The Gnomish Mines" "M" (8, 2)
 ALIGNMENT:	lawful
 DESCRIPTION:	mazelike
-RNDLEVEL:	"minetn" "T" @ (3, 2) 7
+RNDLEVEL:	"m_twn" "T" @ (3, 2) 11
 LEVELDESC:	town
-RNDLEVEL:	"minend" "E" @ (-1, 0) 3
+RNDLEVEL:	"m_end" "E" @ (-1, 0) 4
 
 #
 #	The Questdungeon
diff -ruN nethack-3.4.0./dat/mines.des nethack-3.4.0/dat/mines.des
--- nethack-3.4.0./dat/mines.des	Thu Apr 11 17:58:27 2002
+++ nethack-3.4.0/dat/mines.des	Sun Jun 16 21:31:09 2002
@@ -4,52 +4,119 @@
 # NetHack may be freely redistributed.  See license for details.
 #
 
-#	The "fill" level for the mines.
-#
-#	This level is used to fill out any levels not occupied by
-#	specific levels as defined below.
-#
-MAZE: "minefill" , ' '
-INIT_MAP: '.' , ' ' , true , true , random , true
-NOMAP
-#
-STAIR: random, up
-STAIR: random, down
-#
-OBJECT: '*', random, random
-OBJECT: '*', random, random
-OBJECT: '*', random, random
-OBJECT: '(', random, random
-OBJECT: random, random, random
-OBJECT: random, random, random
-OBJECT: random, random, random
-#
-MONSTER: 'G', "gnome", random
-MONSTER: 'G', "gnome", random
-MONSTER: 'G', "gnome", random
-MONSTER: 'G', "gnome", random
-MONSTER: 'G', "gnome", random
-MONSTER: 'G', "gnome", random
-MONSTER: 'G', "gnome", random
-MONSTER: 'G', "gnome lord", random
-MONSTER: 'h', "dwarf", random
-MONSTER: 'h', "dwarf", random
-MONSTER: 'G', random, random
-MONSTER: 'G', random, random
-MONSTER: 'h', random, random
+# New levels added; changes to Kelly Bailey's original "Grotto Town"
+# were removed.
+#      - Rob Ellwood.
+
+
+# "Orcus Prepares" by Rob Ellwood
+
+#	This is a rehash of the Orcus level.  
 #
-TRAP: random, random
-TRAP: random, random
-TRAP: random, random
-TRAP: random, random
-TRAP: random, random
-TRAP: random, random
+#	With two large general stores, this level is a bit generous.
+MAZE: "m_twn-1",' '
+GEOMETRY:center,center
+
+#          10        20        30        40        50        60        70
+#012345678901234567890123456789012345678901234567890123456789012345678901234
+MAP
+0---------------------------------------------------------------------------
+1|....|      |.........|        |....|....|....|....|.........|....|.......|
+2|....|      |.........|        |....|....|....|....|.........|....|.......|
+3|....+#     |.........|        |....|....|....|--+-|-+-------|....|.......|
+4|....|#     -------+---        |....|....|....|..............+....|.......|
+5------#            #           |....|....|....|..............|....|.......|
+6     -#------------#-------------+-----+----+--....-------...----------+--|
+7     |.............................................|.....|................|
+8     |.............................................|.....|................|
+9     -----#----------------------+---------+---....|.....|...----------+--|
+0          #                    |....|....|....|....---+---...|......|.....|
+1          #                    |....+....|....|..............|......|.....|
+2          #      |------       ----------------....------....+......|.....|
+3          #######+.....+########...................|..| |....|......|.....|
+4 |--|            |.....|       ----------+------+--|..--|....----------+--|
+5 |..|            |.....|       |....|.......|......|..|.+....|............|
+6 |..|            -------       |....+.......|......|..--|....|............|
+7 |--|                          |....|.......|......|..S.|....|............|
+8---------------------------------------------------------------------------
+ENDMAP
+#012345678901234567890123456789012345678901234567890123456789012345678901234
+#          10        20        30        40        50        60        70
+
+RANDOM_PLACES: (03,03),(06,08),(14,02),(20,15)
+
+REGION: (00,00, 30,17), unlit, "ordinary"
+REGION: (31,00, 74,17),   lit, "ordinary"
+
+REGION: (02,15, 03,16), unlit, "vault"
+
+DOOR: closed, (05,03)
+DOOR: closed, (17,13)
+DOOR: closed, (23,13)
+DOOR: closed, (19,04)
+
+STAIR: place[0], up
+STAIR: (63,16) ,down
+
+DOOR: random, (53,03)
+DOOR: random, (61,04)
+DOOR: random, (33,06)
+DOOR: random, (36,11)
+DOOR: closed, (39,06)
+DOOR: closed, (71,06)
+DOOR: open  , (33,09)
+DOOR: random, (43,09)
+DOOR: open  , (71,09)
+DOOR: open  , (41,14)
+DOOR: closed, (71,14)
+DOOR: open  , (56,15)
+DOOR: closed, (36,16)
+
+MONSTER: '@', "watchman", random, peaceful
+MONSTER: '@', "watchman", random, peaceful
+MONSTER: '@', "watchman", random, peaceful
+MONSTER: '@', "watchman", random, peaceful
+MONSTER: '@', "watch captain", random, peaceful
+
+REGION: (52,07,56,09), unlit, "temple"
+ALTAR : (54,08), random, shrine
+DOOR  : closed , (54,10)
+OBJECT: '%', "corpse", (55,07), "gnomish wizard", 0
+OBJECT: '%', "corpse", (54,09), "gnomish wizard", 0
+OBJECT: '%', "corpse", (54,09), "quasit"        , 0
+
+
+REGION: (62,10, 67,13), lit, "shop"
+DOOR  : open  ,(61,12)
+REGION: (42,01, 45,05), lit, "shop"
+DOOR  : open  ,(44,06)
+REGION: (45,15, 50,17), lit, "candle shop"
+DOOR  : open  ,(48,14)
+OBJECT[40%]:')',"long sword", (50,15), random, 0, "Demonbane"
+#  Demonbane is a penalty:  getting good artifacts becomes harder.
+
+MONSTER: 'i', random  , random, hostile
+MONSTER: 'i', random  , random, hostile
+MONSTER: 'i', random  , random, hostile
+MONSTER: 'i', random  , random, peaceful
+MONSTER: 'i', "quasit", random, hostile
+
+
+# Orcus' spy:
+DOOR     : locked, (56,15)
+MONSTER  : '&', "horned devil", (52,13), asleep
+OBJECT   : '(', "stethoscope" , (52,13), cursed, 0
+ENGRAVING:  (53,17), dust, "Orcus was here"
+TRAP     : "hole", (55,17)
+
+WALLIFY
+
 
 
 # Minetown variant 1
 # "Frontier Town"
 #
-LEVEL: "minetn-1"
+LEVEL: "m_twn-2"
 
 ROOM: "ordinary" , lit, (3,3), (center,center), (31,15)
 NAME: "town"
@@ -130,7 +197,7 @@
 # Minetown variant 2
 # "Town Square"
 #
-LEVEL: "minetn-2"
+LEVEL: "m_twn-3"
 ROOM: "ordinary" , lit, (3,3), (center,center), (31,15)
 NAME: "town"
 FOUNTAIN: (17, 5)
@@ -226,7 +293,7 @@
 # Minetown variant 3 by Kelly Bailey
 # "Alley Town"
 #
-LEVEL: "minetn-3"
+LEVEL: "m_twn-4"
 ROOM: "ordinary",lit,(3,3),(center,center),(31,15)
 NAME: "town"
 FOUNTAIN:(01,06)
@@ -317,7 +384,7 @@
 # Minetown variant 4 by Kelly Bailey
 # "College Town"
 #
-LEVEL: "minetn-4"
+LEVEL: "m_twn-5"
 ROOM: "ordinary",lit,(3,3),(center,center),(30,15)
 NAME: "town"
 FOUNTAIN:(08,07)
@@ -398,7 +465,7 @@
 
 # "Grotto Town" by Kelly Bailey
 #
-MAZE: "minetn-5",' '
+MAZE: "m_twn-6",' '
 GEOMETRY:center,center
 MAP
 -----         ---------                                                    
@@ -498,7 +565,7 @@
 
 # "Bustling Town" by Kelly Bailey
 #
-MAZE: "minetn-6",' '
+MAZE: "m_twn-7",' '
 INIT_MAP:'.','-',true,true,lit,true
 GEOMETRY:center,top
 MAP
@@ -571,7 +638,7 @@
 
 # "Bazaar Town" by Kelly Bailey
 #
-LEVEL: "minetn-7"
+LEVEL: "m_twn-8"
 ROOM: "ordinary" , lit, (3,3), (center,center), (30,15)
 NAME: "town"
 FOUNTAIN: (12, 07)
@@ -676,12 +743,533 @@
 MONSTER: 'G', "gnome", random
 
 RANDOM_CORRIDORS
-  
+
+
+# "Ruined & Dangerous Town" by Kelly Bailey
+#
+MAZE: "m_twn-9",' '
+INIT_MAP:'.','-',true,true,unlit,true
+GEOMETRY:center,top
+MAP
+.--------..--.----.--.----..----.
+.|-----...{....---------.----.-|.
+..|...|........|..||...|......|..
+..|...|.------....||...|......||.
+..|...+.|....|....||...|.|....||.
+.||...+.|....|...--|...|.--..--..
+.||...|.|....+.....--+--.......|.
+.||...|.------...........}..}}}|.
+.|-----..........--.+-.....}}}.|.
+.------------------..-----------.
+.................................
+ENDMAP
+ENGRAVING:(19,9),burn,"BEWARE"
+ENGRAVING:(23,7),mark,"DANGER"
+NON_DIGGABLE:(0,0,32,10)
+REGION:(0,0,32,10),unlit,"ordinary"
+REGION:(3,2,5,7),lit,"temple"
+REGION:(9,4,12,6),lit,"candle shop"
+REGION:(16,2,17,4),lit,"food shop"
+REGION:(20,2,22,5),lit,"shop"
+REGION:(26,2,29,4),lit,"tool shop"
+STAIR:(24,1),down
+STAIR:levregion(01,02,75,20),(0,1,32,10),up
+DOOR:open,(20,8)
+DOOR:locked,(6,4)
+DOOR:locked,(6,5)
+DOOR:closed,(13,6)
+DOOR:closed,(21,6)
+ALTAR:(4,2),random,altar
+OBJECT:'`',"boulder",(25,2)
+OBJECT:'`',"boulder",(25,3)
+OBJECT:'`',"boulder",(27,5)
+OBJECT:'`',"boulder",(28,5)
+OBJECT:'`',"boulder",(31,5)
+OBJECT:'`',"boulder",(31,2)
+OBJECT:'`',"boulder",(1,2)
+OBJECT:'`',"boulder",(1,3)
+OBJECT:'`',"boulder",(1,4)
+OBJECT:'`',"boulder",(9,0)
+OBJECT:'`',"boulder",(10,0)
+OBJECT:'`',"boulder",(13,0)
+OBJECT:'`',"boulder",(18,0)
+OBJECT:'`',"boulder",(21,0)
+OBJECT:'`',"boulder",(26,0)
+OBJECT:'`',"boulder",(27,0)
+OBJECT:'`',"boulder",(15,3)
+OBJECT:'`',"boulder",(15,4)
+OBJECT:'`',"boulder",(15,5)
+OBJECT:'`',"boulder",(16,5)
+OBJECT:'`',"boulder",(16,8)
+OBJECT:'`',"boulder",(19,8)
+OBJECT:'`',"boulder",(20,9)
+OBJECT:'%',"corpse",(27,2),"shopkeeper",0
+OBJECT:'(',random,(28,2)
+OBJECT:'(',random,(28,2)
+CONTAINER:'`',"statue",(30,8),uncursed,"gnome king",0
+OBJECT[50%]:'(',"bag of holding",contained,cursed,1
+MONSTER: 'G', "gnome", (3,2), hostile
+MONSTER: 'G', "gnome", (4,2), hostile
+MONSTER: 'G', "gnome", (5,2), hostile
+MONSTER: 'G', "gnome", (3,3), hostile
+MONSTER: 'G', "gnome", (4,3), hostile
+MONSTER: 'G', "gnome", (5,3), hostile
+MONSTER: 'G', "gnome", (3,4), hostile
+MONSTER: 'G', "gnome", (4,4), hostile
+MONSTER: 'G', "gnome", (5,4), hostile
+MONSTER: 'G', "gnomish wizard",(3,5), hostile
+MONSTER: 'G', "gnomish wizard",(4,5), hostile
+MONSTER: 'G', "gnome king",(5,5), hostile
+MONSTER: 'G', "gnome lord", random, hostile
+MONSTER:';',"giant eel",(29,7), hostile
+MONSTER:'D',"white dragon",(29,2), hostile
+
+NOMAP
+MONSTER:'o',random,random
+MONSTER:'o',random,random
+MONSTER:'o',random,random
+MONSTER:'h',random,random
+MONSTER:'h',random,random
+OBJECT:')',random,random
+OBJECT:')',random,random
+OBJECT:'(',random,random
+OBJECT:'(',random,random
+
+
+# "Mini-Castle Town" by Kelly Bailey
+# 
+MAZE:"m_twn-10",' '
+INIT_MAP:'.','-',true,true,lit,true
+GEOMETRY:center,center
+MAP
+----...}}}}}}}}}}}}}}}}}}}}}}}.--.---
+--.....}---------------------}..-..--
+-.----.}|........|.....|..|.|}......-
+..|..|.}|.-----..|.....S.K|.|}.......
+..|..|.}|.|...|..|.....|..|.|}.......
+..|..|.}|.|...+..---+------.|}.......
+..--+-.}|.|...|.............|}......-
+.......}|.-----..T...{...T..|}......-
+..{....}|...................|}.......
+.......}|.--+--.-----.-----.|}.......
+.......}|.|...--|...|.|...|.|}.......
+.......}|.|...S.|...+.+...|.|}.......
+..-....}|.|...|.|...|.|...|.|}.-+--..
+---....}|.-----------.-----.|}.|..|..
+-......}|...................|}.|....-
+--..-..}---------------------}......-
+------.}}}}}}}}}}}}}}}}}}}}}}}..-----
+ENDMAP
+
+REGION:(0,0,36,16),lit,"ordinary"
+NON_DIGGABLE:(8,1,28,15)
+STAIR:levregion(1,1,20,20),(0,0,36,16),down
+STAIR:levregion(60,1,78,20),(0,0,36,16),up
+REGION:(3,3,4,5),lit,"food shop"
+REGION:(11,4,13,6),lit,"candle shop"
+REGION:(18,2,22,4),lit,"temple"
+ALTAR:(20,3),align[1],shrine
+REGION:(24,2,25,4),lit,"ordinary"
+REGION:(11,10,12,12),lit,"ordinary"
+REGION:(14,11,15,12),unlit,"ordinary"
+REGION:(17,10,19,12),lit,"tool shop"
+REGION:(23,10,25,12),lit,"shop"
+DOOR:closed,(4,6)
+DOOR:closed,(14,5)
+DOOR:closed,(20,5)
+DOOR:locked,(23,3)
+DOOR:locked,(12,9)
+DOOR:locked,(14,11)
+DOOR:closed,(20,11)
+DOOR:closed,(22,11)
+DOOR:locked,(32,12)
+DRAWBRIDGE:(7,8),east,open
+DRAWBRIDGE:(29,8),west,open
+MONSTER:'@',"watchman",(3,8),peaceful
+MONSTER:'@',"watchman",(13,8),peaceful
+MONSTER:'@',"watchman",(23,8),peaceful
+MONSTER:'@',"watch captain",(18,7),peaceful,asleep
+MONSTER:'G',"gnomish wizard",(25,3)
+MONSTER:'G',"gnomish wizard",(24,3)
+MONSTER:'G',"gnome lord",(11,12)
+MONSTER:'G',"gnome",(12,12)
+MONSTER:'G',"gnome",(11,11)
+MONSTER:'Z',"skeleton",(15,12)
+MONSTER:'O',"ogre",(33,13),asleep
+MONSTER:'O',"ogre",(32,13),asleep
+MONSTER:'O',"ogre",(33,14),asleep
+OBJECT:'`',"boulder",(34,14)
+NOMAP
+MONSTER:'G',"gnome",random
+MONSTER:'h',"dwarf",random
+MONSTER:'h',"dwarf lord",random
+MONSTER:'G',"gnome lord",random
+
+
+#  "Conquered Town"  by Rob Ellwood  
+#  The NetHack license applies; freely redistribute.
+#  Version:  2002, March 20, 2:09 PM
+
+MAZE: "m_twn-11", ' '
+
+MESSAGE: "You see soldiers with nametags."
+#  A rationalization for why you know their names.
+MESSAGE: "The soldier named Rat Face shouts:  'Intruder!  KILL!'"
+
+
+GEOMETRY:center,center
+
+#12345678901234567890123456789012345678901234567890123456789012345678901234
+MAP
+--------  --------         -----------     ---------                      
+|-|....|  |....|-|         |...F.F...|     |.S..K..|                  ---  
+|.S....|  |....S.| |-------|...+.+...|     |-|.....|                  |.| 
+--|---+----+---|-- |.......|...F.F...|       |.....|                  ---
+  |............|   |.......|----.------------|--+--|                       
+  |............|   |.......|...F.F...|...|...|.....|                       
+  |............|   |.......|...+.+...|...|...|.....|                       
+  |............|   |.......|...F.F...|...|...|.....|                       
+  |............|-------+--------.------+---+----+-------------------------|
+  |............+...+.................+....................................|
+  |............+...+.................+....................................|
+  |............+...+.................+....................................|
+  |............|-------+--------.------+---+----+------.--------+------+--|
+  |............|   |.......|...|.|...|...|...|.+...| .....  |........|....|
+  |............|   |.......|...+.+...|...|...|--..{|  ....  |........|....|
+  |............|   |.......|...|.|...|...|...|.+..K|  ..... |........|....|
+  |............|   |.......|----.----|-------|----P|  ..... |........|....|
+--|---+----+---|-- |.......|...|.|...|           |P|    ..  |--------|----|
+|.S....|  |....S.| |-------|...+.+...|           |P|        
+|-|....|  |....|-|         |...|.|...|            . 
+--------  --------         -----------              
+ENDMAP
+#12345678901234567890123456789012345678901234567890123456789012345678901234
+#  Ambush area     Equipment  Cells  Brrcks   Mess                    Sig
+#     "    "       Equipment   BOQ   Brrcks   Bthrm  Cnstrctn Shrine  Strs
+
+
+#  If you first enter this level by any way except the staircase, the
+#  initial message about soldiers with nametags makes no sense.  Fix:
+#  if you skip the stairs, you end up in the ambush area.
+
+TELEPORT_REGION:(03,04,14,16),(0,0,0,0),down
+TELEPORT_REGION:(03,04,14,16),(0,0,0,0),up
+
+
+# Ambush area.
+
+REGION : (03,04,14,16), lit, "ordinary"
+STAIR  : (08,10),up
+
+MONSTER: '@',"soldier", (06,04), "Chumley"
+MONSTER: '@',"soldier", (11,04), "Rat Face"
+MONSTER: '@',"soldier", (04,18), "Killer"     , asleep
+MONSTER: '@',"soldier", (12,18), "Naughty Boy", asleep
+
+OBJECT: '!', "invisibility", (06,04)
+OBJECT: '!', "invisibility", (11,04)
+OBJECT: '!', "invisibility", (06,18)
+OBJECT: '!', "invisibility", (12,18)
+
+#	Rooms for the four ambushers.
+REGION: (03,01, 06,02),   lit, "ordinary"
+DOOR:   open,   (06,03)
+REGION: (01,02, 01,02), unlit, "ordinary"
+DOOR:   closed, (02,02)
+OBJECT[50%]: '/', "secret door detection", (01,02)
+OBJECT[50%]: random, random, (01,02)
+
+REGION: (11,01, 14,02),   lit, "ordinary"
+DOOR:   open  , (11,03)
+REGION: (16,02, 16,02), unlit, "ordinary"
+DOOR:   closed, (15,02)
+OBJECT[50%]: ')', "silver dagger", (16,02)
+OBJECT[50%]: random, random, (16,02)
+
+REGION: (03,18, 06,19),   lit, "ordinary"
+DOOR:   open,  (06,17)
+REGION: (01,18, 01,18), unlit, "ordinary"
+DOOR:   closed,(02,18)
+OBJECT[50%]: '"', "amulet of restful sleep", (01,18), blessed,0
+OBJECT[50%]: random, random, (01,18)
+
+REGION     :  (11,18, 14,19), unlit, "ordinary"
+DOOR       :  locked, (11,17)
+MONSTER    :  'n',  "water nymph", (13,19), asleep, "Mistress Dominique"
+OBJECT     :  '[',  "studded leather armor",(13,19)
+OBJECT     :  '[',  "high boots",           (13,19)
+OBJECT[50%]:  ')',  "bullwhip",             (13,19)
+REGION     :  (16,18, 16,18), unlit, "ordinary"
+#  Closet:
+DOOR       :  locked, (15,18)
+OBJECT[50%]:  '(',  "leash",                (16,18)
+OBJECT[50%]:  '(',  "blindfold",            (16,18)
+OBJECT     :  '_',  "iron chain",           (16,18)
+OBJECT     :  '_',  "iron chain",           (16,18)
+OBJECT     :  '_',  "iron chain",           (16,18)
+OBJECT     :  '_',  "iron chain",           (16,18)
+OBJECT     :  '(',  "wax candle",           (16,18)
+OBJECT     :  '(',  "wax candle",           (16,18)
+OBJECT     :  '(',  "wax candle",           (16,18)
+OBJECT     :  '(',  "wax candle",           (16,18)
+OBJECT[50%]:  '?',  "punishment",           (16,18)
+OBJECT[50%]:  '/',  "probing",              (16,18)
+#   You get 1-7 (?) candles each time.  I ask for four sets of candles to give
+#   good odds that you get 7+.
+MONSTER: 'Z', "skeleton",             (16,18)
+
+DOOR: locked, (15,09)
+DOOR: locked, (15,10)
+DOOR: locked, (15,11)
+
+#  Murdered townsmen.  Good loot was collected.  Izchak escaped.
+#  Put the corpses last, so they're on the top of the heap.
+
+OBJECT[50%]:  '(', "skeleton key", (03,04)
+OBJECT[50%]:  '(', "skeleton key", (03,04)
+OBJECT[50%]:  '(', "skeleton key", (03,04)
+OBJECT[50%]:  '/', "fire"        , (03,04), uncursed  , 0
+OBJECT[50%]:  '/', "cold"        , (03,04), uncursed  , 0
+OBJECT[50%]:  '/', "striking"    , (03,04), uncursed  , 0
+OBJECT[50%]:  '/', "lightning"   , (03,04), uncursed  , 0
+OBJECT[20%]:  '/', "death"       , (03,04), uncursed  , 0
+OBJECT[10%]:  random, random     , (03,04)
+OBJECT     :  '%', "corpse"      , (03,04), "watchman", 0
+OBJECT     :  '%', "corpse"      , (03,04), "priest"  , 0
+OBJECT[15%]:  '[', "robe"        , (03,04)
+
+OBJECT[90%]:  '%', "corpse", (03,04), uncursed, "shopkeeper", 0, "Pakka Pakka"
+OBJECT[90%]:  '%', "corpse", (03,04), uncursed, "shopkeeper", 0, "Tjisolok"
+OBJECT[90%]:  '%', "corpse", (03,04), uncursed, "shopkeeper", 0, "Y-crad"
+
+MONSTER[15%]:'Z',"human zombie",(03,04), peaceful, "Manlobbi"
+
+OBJECT     :  '%', "corpse"   , (25,09), "watchman"      ,0
+OBJECT     :  '%', "corpse"   , (34,02), "watchman"      ,0
+OBJECT     :  '%', "corpse"   , (11,10), "watch captain" ,0
+
+
+
+# Entrance room.
+
+REGION: (16,09, 18,11), lit, "ordinary"
+
+ENGRAVING: (17,10), engrave, "Our Leader"
+CONTAINER: '`', "statue", (17,10), "titan", 0, "The Wizard of Yendor"
+#  Titan?  The sculptor obviously took a few liberties with reality.
+OBJECT   : '"', "cheap plastic imitation of the Amulet of Yendor", contained, cursed, 0
+
+DOOR: closed, (19,09)
+DOOR: closed, (19,10)
+DOOR: closed, (19,11)
+
+REGION: (20,09, 73,11), lit, "ordinary"
+
+
+#	Prison cells.
+
+ENGRAVING:  (32,08), engrave, "Cells"
+
+REGION: (28,01, 30,03),   lit, "ordinary"
+REGION: (28,05, 30,07), unlit, "ordinary"
+REGION: (34,01, 36,03),   lit, "ordinary"
+REGION: (34,05, 36,07),   lit, "ordinary"
+
+DOOR: locked,(31,02)
+DOOR: locked,(31,06)
+DOOR: locked,(33,02)
+DOOR: locked,(33,06)
+
+#  One watchman surrendered and was locked up.
+MONSTER: '@', "watchman", (35,06), peaceful
+
+
+#	Shops. 
+
+REGION: (20,03, 26,07), lit, "armor shop"
+DOOR:   open, (23,08)
+REGION: (20,13, 26,17), lit, "weapon shop"
+DOOR:   open, (23,12)
+
+#	Unused rooms.
+
+ENGRAVING:    (32,12), engrave, "BOQ"
+DOOR: random, (31,14)
+DOOR: random, (31,18)
+DOOR: random, (33,14)
+DOOR: random, (33,18)
+
+OBJECT: '('   , "chest", (29, 18)
+OBJECT: random, random , (34, 17)
+
+
+#     Entrance to the inner area:
+
+ENGRAVING:  (36,09), engrave, "Authorized personnel only"
+ENGRAVING:  (36,10), engrave, "FDF Advance Base 3"
+ENGRAVING:  (36,11), engrave, "Authorized personnel only"
+#  FDF = Frisian Defense Force, SF, David Drake.
+
+DOOR: closed, (37,09)
+DOOR: closed, (37,10)
+DOOR: closed, (37,11)
+
+REGION: (28,13, 30,15), unlit, "ordinary"
+REGION: (28,17, 30,19), unlit, "ordinary"
+REGION: (34,13, 36,15), unlit, "ordinary"
+REGION: (34,17, 36,19), unlit, "ordinary"
+
+
+# Mess hall.
+
+ENGRAVING: (48,09), engrave, "Mess hall"
+REGION   : (46,05,50,07), lit, "ordinary"
+DOOR     : closed, (48,08)
+
+OBJECT[60%]: '%', "tripe ration", (47,05), uncursed, 0, "Alpo"
+OBJECT[60%]: '%', "tripe ration", (47,06), uncursed, 0, "butcher shop scraps"
+OBJECT[60%]: '%', "tripe ration", (47,07), uncursed, 0, "Gravy Train"
+OBJECT[60%]: '%', "tripe ration", (49,05), uncursed, 0, "Kibbles and Bits"
+OBJECT[60%]: '%', "tripe ration", (49,06), uncursed, 0, "Purina"
+OBJECT[60%]: '%', "tripe ration", (49,07), uncursed, 0, "Doggie Treats"
+
+
+# Kitchen:  the map has a sink.
+
+REGION : (46,01,50,03), lit, "ordinary"
+DOOR   : locked, (48,04)
+MONSTER: '@', "sergeant", (48,02), peaceful, "Cookie"
+OBJECT : '[', "alchemy smock", (48,02), uncursed, 0, "Iron chef"
+
+#  A fire pit as the oven:
+TRAP  : "fire", (50, 02)
+OBJECT: '`', "boulder", (50,01)
+OBJECT: '`', "boulder", (50,03)
+OBJECT: '`', "boulder", (49,01)
+OBJECT: '`', "boulder", (49,03)
+
+#  Cookie's hidden stash:
+DOOR       : locked, (45,01)
+CONTAINER  : '(', "chest", (44,01)
+OBJECT[60%]: '!', "booze", contained, cursed, 0, "Screech"
+OBJECT[60%]: '!', "booze", contained, cursed, 0, "Old Rotgut"
+OBJECT[60%]: '!', "booze", contained, cursed, 0, "paint remover"
+OBJECT[60%]: '!', "booze", contained, cursed, 0, "meth"
+OBJECT[60%]: '!', "booze", contained, cursed, 0, "Sterno"
+OBJECT[10%]: '!', "booze", contained, blessed, 0, "Guinness"
+
+
+# Bathroom.  The soldiers will slowly vanish down the holes once the 
+# doors are unlocked.  Oh, well.
+
+REGION: (46, 13, 50, 16), lit, "ordinary"
+DOOR  : closed, (48,12)
+TRAP  : "hole", (46, 13)
+TRAP  : "hole", (46, 15)
+DOOR  : locked, (47, 13)
+DOOR  : locked, (47, 15)
+
+#  A fountain as a water supply, then a sink with:
+ENGRAVING: (50,15), engrave, "Soldiers on KP must wash hands"
+#  ...the water overflows and is sewered:
+TRAP     : "hole", (50,19)
+#  Shower:
+TRAP     : "rust", (50,13)
+
+#  Peaceful soldiers.  If you're past the entranceway, you must belong
+#  here, right?  They can't be placed at random, or the ambush would 
+#  look wrong.  The MPs are guardsmen, and will become angry if they see
+#  players unlocking doors.  
+
+MONSTER: '@',"soldier", (23, 03), peaceful, "Raoul"
+MONSTER: '@',"soldier", (20, 10), peaceful, "Adamski"
+MONSTER: '@',"soldier", (46, 05), peaceful, asleep, "Boozer"
+MONSTER: '@',"soldier", (47, 05), peaceful, "Sven"
+MONSTER: '@',"soldier", (46, 06), peaceful, "Tac"
+
+MONSTER: '@', "watchman", (32, 07), peaceful, asleep, "MP Richards"
+MONSTER: '@', "watchman", (23, 16), peaceful, "MP Hodd"
+MONSTER: '@', "watchman", (24, 16), peaceful, "MP Gilbert"
+MONSTER: '@', "watchman", (42, 10), peaceful, "MP Zazzi"
+
+
+# Four barracks.  Two shifts are off duty & asleep right now.  Making
+# the soldiers hostile makes the level too tough.  Instead, make the
+# leaders hostile:  "Hey!  You don't belong here!"  Put them at the back
+# of the rooms.  There's the initial battle, a long delay, then enough
+# soldiers wake up to let the leaders out.
+# Problem:  one bugle and the player is hit by 21 attackers.  Fix:  no
+# proper sargeants or officers.
+
+REGION: (38, 05, 40, 07),   lit, "ordinary"
+REGION: (42, 05, 44, 07),   lit, "ordinary"
+REGION: (38, 13, 40, 16), unlit, "ordinary"
+REGION: (42, 13, 45, 16), unlit, "ordinary"
+
+DOOR  : open  , (39, 08)
+DOOR  : open  , (43, 08)
+DOOR  : closed, (39, 12)
+DOOR  : closed, (43, 12)
+
+ENGRAVING: (39,09), engrave, "Squad A"
+ENGRAVING: (43,09), engrave, "Squad B"
+ENGRAVING: (39,11), engrave, "Squad C"
+ENGRAVING: (43,11), engrave, "Squad D"
+
+MONSTER: '@',"soldier" , (38, 14), peaceful, asleep
+MONSTER: '@',"soldier" , (39, 14), peaceful, asleep
+MONSTER: '@',"soldier" , (40, 14), peaceful, asleep
+MONSTER: '@',"soldier" , (38, 15), peaceful, asleep
+MONSTER: '@',"soldier" , (39, 15), peaceful, asleep
+MONSTER: '@',"soldier" , (40, 15),           asleep, "Sarge"
+
+MONSTER: '@',"soldier" , (42, 14), peaceful, asleep
+MONSTER: '@',"soldier" , (43, 14), peaceful, asleep
+MONSTER: '@',"soldier" , (44, 14), peaceful, asleep
+MONSTER: '@',"soldier" , (42, 15), peaceful, asleep
+MONSTER: '@',"soldier" , (43, 15), peaceful, asleep
+MONSTER: '@',"soldier" , (44, 15),           asleep, "Top"
+
+
+#  New room, partly carved out.
+
+ENGRAVING: (55, 13), dust, "Hard hat area"
+OBJECT: '`', "boulder", (54,15)
+OBJECT: '*', "rock", (57,17)
+OBJECT: '*', "rock", (57,17)
+
+
+# Temple of Moloch.
+
+REGION:(61,13,68,16), unlit, "temple"
+ALTAR :(64,15), noalign, shrine
+DOOR  : random, (64,12)
+
+
+# Stairs down.  Richard Nixon & Spiro Agnew.
+
+REGION: (70, 13, 73, 16), lit, "ordinary"
+DOOR  : locked, (71, 12)
+STAIR : (71, 15), down
+MONSTER: '@',"soldier", (72,14), "Nix"
+MONSTER: '@',"soldier", (71,14), "Aggenu"
+
+
+# Author's signature.  Dwarf is the closest thing to an engineer.
+
+OBJECT   : '%', "corpse", (71, 02), "dwarf", 0
+TRAP     : "teleport"   , (71, 02)
+ENGRAVING: (71, 02), mark, "I wuz Robbed"
+#  Hey!  Who took his Magic Marker?
+
+
+
+
 
 # Mine end level variant 1
 # "Mimic of the Mines"
 #
-MAZE: "minend-1", ' '
+MAZE: "m_end-1", ' '
 GEOMETRY:center,center
 #1234567890123456789012345678901234567890123456789012345678901234567890
 MAP
@@ -792,7 +1380,7 @@
 # Mine end level variant 2
 # "Gnome King's Wine Cellar"
 #
-MAZE: "minend-2", ' '
+MAZE: "m_end-2", ' '
 GEOMETRY:center,center
 MAP
 ---------------------------------------------------------------------------
@@ -911,7 +1499,7 @@
 # "Catacombs" by Kelly Bailey
 # Relies on some very specific behavior of MAZEWALK.
 #
-MAZE:"minend-3",'-'
+MAZE:"m_end-3",'-'
 FLAGS:nommap
 GEOMETRY:center,bottom
 MAP
@@ -1005,4 +1593,194 @@
 MONSTER:'e',random,random
 
 
+# "Boulder Bonanza" by Kelly Bailey
+#
+MAZE: "m_end-4", ' '
+GEOMETRY:center,center
+MAP
+                        ---------  ----                        
+                ---------.......-.--..--   ---                 
+                -..........----S---....-- --.--       ---      
+    ----------  --.....-----  -.........---...-       -.---    
+  ---........--  -...---     --....-...--.....----   --...---  
+ --...........-- -...-     ---....--.............--  -......-- 
+--.....---.....---...-     -.....--...............- --....---  
+-......--..........---     --..........----.....--- -.....-    
+-..................-      --...........-  -....--   -.....--   
+--.............-----      -............-- --....-  --......-   
+ --..........---     ------....----.....- .--...----.......-   
+  --........--      --.........-  ---...----..............--   
+   ---...----       -...-......-    -.....................-    
+     -----          -..---....--    --.......---..........-    
+                    -...--...--      -....---- --.......---    
+                    --......--     ---...--     --.....--      
+                     --------      -.....-       -------       
+                                   --...--                     
+                                    -----                      
+ENDMAP
+RANDOM_PLACES:(1,7),(3,6),(4,5),(6,4),(7,9),(10,9),(12,5),(13,8),(14,7),(16,8)
+
+NON_DIGGABLE:(00,00,62,18)
+
+STAIR:(56,04),up
+DOOR:locked,(31,02)
+
+TRAP:"rolling boulder",random
+TRAP:"rolling boulder",random
+TRAP:"rolling boulder",random
+TRAP:random,random
+TRAP:random,random
+TRAP:random,random
+TRAP:"falling rock",random
+TRAP:"falling rock",random
+TRAP:"falling rock",random
+TRAP:"falling rock",random
+TRAP:"board",(17,07)
+# Mindless critters
+MONSTER:'Z',random,random
+MONSTER:'Z',random,random
+MONSTER:'Z',random,random
+MONSTER:'Z',random,random
+MONSTER:'Z',random,random
+MONSTER:'Z',random,random
+MONSTER:'Z',random,random
+MONSTER:'Z',random,random
+MONSTER:'Z',random,random
+MONSTER:'Z',random,random
+MONSTER:'Z',random,(42,10)
+MONSTER:''',random,place[0],asleep
+MONSTER:''',random,place[1],asleep
+MONSTER:''',random,place[2],asleep
+MONSTER:''',random,place[3],asleep
+MONSTER:''',random,place[4],asleep
+MONSTER:''',random,place[5],asleep
+MONSTER:''',"stone golem",place[6],asleep
+MONSTER:''',random,place[7],asleep
+MONSTER:''',random,place[8],asleep
+MONSTER:''',random,place[9],asleep
+# Many, many boulders and rocks
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'`',"boulder",random
+OBJECT:'*',"flint",(01,08)
+OBJECT:'*',"rock",(02,09)
+OBJECT:'*',"flint",(03,05)
+OBJECT:'*',"rock",(04,11)
+OBJECT:'*',"loadstone",(05,10)
+OBJECT:'*',"rock",(06,08)
+OBJECT:'*',"flint",(07,12)
+OBJECT:'*',"flint",(08,11)
+OBJECT:'*',"rock",(09,07)
+OBJECT:'*',"loadstone",(10,10)
+OBJECT:'*',"rock",(11,11)
+OBJECT:'*',"flint",(12,09)
+OBJECT:'*',"rock",(13,06)
+OBJECT:'*',"loadstone",(14,09)
+OBJECT:'*',"rock",(15,08)
+OBJECT:'*',"flint",(16,07)
+OBJECT:'*',"loadstone",(04,10)
+OBJECT:'*',"rock",(09,10)
+OBJECT:'*',"flint",(12,10)
+OBJECT:'*',"rock",(07,04)
+OBJECT:'*',"loadstone",(08,05)
+OBJECT:'*',"rock",(10,04)
+OBJECT:'*',"diamond",place[0]
+OBJECT:'*',random,place[1]
+OBJECT:'*',"diamond",place[2]
+OBJECT:'*',random,place[3]
+OBJECT:'*',"emerald",place[4]
+OBJECT:'*',random,place[5]
+OBJECT:'*',"emerald",place[6]
+OBJECT:'*',random,place[7]
+OBJECT:'*',"emerald",place[8]
+OBJECT:'*',random,place[9]
+OBJECT:'*',"ruby",place[0]
+OBJECT:'*',random,place[1]
+OBJECT:'*',"ruby",place[2]
+OBJECT:'*',"amethyst",place[3]
+OBJECT:'*',random,place[4]
+OBJECT:'*',"amethyst",place[5]
+OBJECT:'*',"luckstone",place[6],blessed,0
+OBJECT:'*',random,place[7]
+OBJECT:'*',random,place[8]
+OBJECT:'*',random,place[9]
+OBJECT:'*',"flint",place[0]
+OBJECT:'*',"loadstone",place[1]
+OBJECT:'*',"flint",place[2]
+OBJECT:'*',"flint",place[3]
+OBJECT:'*',"loadstone",place[4]
+OBJECT:'*',"loadstone",place[5]
+OBJECT:'*',"flint",place[6]
+OBJECT:'*',"loadstone",place[7]
+OBJECT:'*',"loadstone",place[8]
+OBJECT:'*',"flint",place[9]
+OBJECT:'(',"magic marker",(42,10)
+OBJECT:'(',random,random
+OBJECT:'(',random,random
+OBJECT:random,random,random
+OBJECT:random,random,random
+OBJECT:random,random,random
+
+
+
+#	The "fill" level for the mines.
+#
+#	This level is used to fill out any levels not occupied by
+#	specific levels as defined below.
+#
+MAZE: "minefill" , ' '
+INIT_MAP: '.' , ' ' , true , true , random , true
+NOMAP
+#
+STAIR: random, up
+STAIR: random, down
+#
+OBJECT: '*', random, random
+OBJECT: '*', random, random
+OBJECT: '*', random, random
+OBJECT: '(', random, random
+OBJECT: random, random, random
+OBJECT: random, random, random
+OBJECT: random, random, random
+#
+MONSTER: 'G', "gnome", random
+MONSTER: 'G', "gnome", random
+MONSTER: 'G', "gnome", random
+MONSTER: 'G', "gnome", random
+MONSTER: 'G', "gnome", random
+MONSTER: 'G', "gnome", random
+MONSTER: 'G', "gnome", random
+MONSTER: 'G', "gnome lord", random
+MONSTER: 'h', "dwarf", random
+MONSTER: 'h', "dwarf", random
+MONSTER: 'G', random, random
+MONSTER: 'G', random, random
+MONSTER: 'h', random, random
+#
+TRAP: random, random
+TRAP: random, random
+TRAP: random, random
+TRAP: random, random
+TRAP: random, random
+TRAP: random, random
+
+
 # end mines.des
diff -ruN nethack-3.4.0./dat/rnd1.des nethack-3.4.0/dat/rnd1.des
--- nethack-3.4.0./dat/rnd1.des	Thu Jan  1 01:00:00 1970
+++ nethack-3.4.0/dat/rnd1.des	Sun Jun 16 21:31:09 2002
@@ -0,0 +1,1194 @@
+#  The header block is at the end of the file.
+
+#  "Mini-bigroom", by Rob Ellwood
+
+MAZE:"rnd1-7",' '
+FLAGS: shortsighted
+GEOMETRY:center,center
+
+#  I don't want to be too hard on low-level characters.  "Shortsighted"
+#  keeps the monsters from piling up on the player.  There are fewer monsters
+#  than the usual bigroom.  Many of them are peaceful or asleep.  I took
+#  out 'O' (ogres); an ogre lord with a battleax is too tough.
+
+#          10        20        30        40        50        60        70
+#012345678901234567890123456789012345678901234567890123456789012345678901234
+MAP
+0------------------------------------- -------------------------------------
+1|...................................---...................................|
+2|.........................................................................|
+3|.........................................................................|
+4|.................................|-----|.................................|
+5|.................................|.....|.................................|
+6|........---------------------------------------------------------........|
+7|........|...............|...............|...............|................|
+8|........|...............|...............|...............|................|
+9|........|...............|...............|...............|................|
+0|........|.......|.......|.......|.......|.......|.......|.......|........|
+1|........|.......|.......|.......|.......|.......|.......|.......|........|
+2|........|.......|.......|.......|.......|.......|.......|.......|........|
+3|........|.......|.......|.......|.......|.......|.......|.......|........|
+4|........|.......|.......|.......|.......|.......|.......|.......|........|
+5|........|.......|.......|.......|.......|.......|.......|.......|........|
+6|................|...............|...............|...............|........|
+7|................|...............|...............|...............|........|
+8|................|...............|...............|...............|........|
+9---------------------------------------------------------------------------
+ENDMAP
+#          10        20        30        40        50        60        70
+#012345678901234567890123456789012345678901234567890123456789012345678901234
+
+RANDOM_OBJECTS:  '+', '$', '$', '?', '=', '!', '(', ')', '['
+RANDOM_MONSTERS: 'h', 'h', 'h', 'n', 'r', 'C', 'K', 'Y', 'Y'
+
+
+REGION: (01,01, 73,18), lit, "ordinary"
+STAIR : random, up
+STAIR : random, down
+
+OBJECT [50%]: object[0], random, random
+OBJECT [50%]: object[0], random, random
+OBJECT [50%]: object[0], random, random
+OBJECT [50%]: object[0], random, random
+OBJECT [50%]: object[0], random, random
+OBJECT [50%]: object[0], random, random
+OBJECT [50%]: object[0], random, random
+OBJECT [50%]: object[0], random, random
+OBJECT [50%]: object[1], random, random
+OBJECT [50%]: object[1], random, random
+OBJECT [50%]: object[1], random, random
+OBJECT [50%]: object[1], random, random
+OBJECT [50%]: random,    random, random
+OBJECT [50%]: random,    random, random
+OBJECT [50%]: random,    random, random
+OBJECT [50%]: random,    random, random
+GOLD:   random,    random
+GOLD:   random,    random
+GOLD:   random,    random
+GOLD:   random,    random
+GOLD:   random,    random
+GOLD:   random,    random
+
+TRAP: random, random
+TRAP: random, random
+TRAP: random, random
+TRAP: random, random
+TRAP: random, random
+TRAP: random, random
+
+MONSTER [50%]: monster[0], random, random
+MONSTER [50%]: monster[0], random, random
+MONSTER [50%]: monster[0], random, random
+MONSTER [50%]: monster[0], random, random
+MONSTER [50%]: monster[0], random, random
+MONSTER [50%]: monster[0], random, random
+MONSTER [50%]: monster[0], random, random
+MONSTER [50%]: monster[0], random, random, asleep
+MONSTER [50%]: monster[0], random, random, asleep
+MONSTER [50%]: monster[0], random, random, peaceful
+
+MONSTER [50%]: monster[1], random, random
+MONSTER [50%]: monster[1], random, random
+MONSTER [50%]: monster[1], random, random, peaceful
+MONSTER [50%]: monster[1], random, random, peaceful
+MONSTER [50%]: monster[1], random, random, peaceful
+MONSTER [50%]: monster[1], random, random, peaceful
+MONSTER [50%]: monster[1], random, random, peaceful
+MONSTER [50%]: monster[1], random, random, peaceful
+
+MONSTER [50%]: monster[2], random, random, asleep
+MONSTER [50%]: monster[2], random, random, asleep
+MONSTER [50%]: monster[2], random, random, asleep
+MONSTER [50%]: monster[2], random, random, asleep
+MONSTER [50%]: monster[2], random, random, asleep
+MONSTER [50%]: monster[2], random, random, asleep
+MONSTER [50%]: monster[2], random, random
+MONSTER [50%]: monster[2], random, random
+
+MONSTER [50%]: random, random, random
+MONSTER [50%]: random, random, random
+MONSTER [50%]: random, random, random
+MONSTER [50%]: random, random, random
+MONSTER [50%]: random, random, random, asleep
+MONSTER [50%]: random, random, random, asleep
+MONSTER [50%]: random, random, random, peaceful
+MONSTER [50%]: random, random, random, peaceful
+
+MONSTER [50%]: 'F'   , random, random
+MONSTER [50%]: 'F'   , random, random
+MONSTER [50%]: 'F'   , random, random
+MONSTER [50%]: 'F'   , random, random
+MONSTER [50%]: 'F'   , random, random
+MONSTER [50%]: 'F'   , random, random
+
+
+#  "Half-Assed Castle", by Rob Ellwood
+
+#	Goal:  YASD players who go straight in and fight before
+#  they're strong enough.  Let players easily bypass the fight.
+#  This is a pathetic imitation of the Castle.
+
+MAZE    : "rnd1-1", ' '
+INIT_MAP: '.' , ' ' , true , true , lit , false
+GEOMETRY: right, center
+
+#          10        20        30        40    
+#0123456789012345678901234567890123456789012345
+MAP
+0..............................................
+1.......}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}.
+2.......}------------------------------------}.
+3.......}|..................................|}.
+4.......}-----------------S------------------}.
+5.......}F......|.....+.......|.....+.+.....|}.
+6.......}F......-------.......|------+-------}.
+7........S......+.....+.......S.............S..
+8.......}F......-------.......|------+-------}.
+9.......}F......|.....+.......|.....+.+.....|}.
+0.......}-----------------+------------------}.
+1.......}|..................................|}.
+2.......}------------------------------------}.
+3.......}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}.
+4..............................................
+ENDMAP
+#0123456789012345678901234567890123456789012345
+#          10        20        30        40    
+
+RANDOM_PLACES : (10,03), (09,11), (42,03), (42,11)
+RANDOM_OBJECTS: '%', '[', ')', '(', '$'
+
+RANDOM_MONSTERS: 'Y', 'O', 'C', 'h', 'Y', 'O', 'C', 'h', 'K'
+REGION: (00,00, 44,14), lit, "ordinary"
+
+#  No arrivals inside the castle:
+TELEPORT_REGION: levregion(01,00, 20,20), (7,1, 44,13)
+STAIR          : levregion(01,00, 20,20), (7,1, 44,13), up
+STAIR: (42,07), down
+
+#  Moat monsters.  Two is enough, I think.  (Snakes are
+#  no good:  loot is added beside the water, so they
+#  have something to hide under.)
+MONSTER: ':', "baby crocodile", (25,01)
+MONSTER: ':', "baby crocodile", (25,13)
+
+#  Entrance room:
+MONSTER: monster[0], random, (09,05)
+MONSTER: monster[0], random, (14,09)
+MONSTER: monster[0], random, (09,07), asleep
+MONSTER: monster[1], random, (09,09)
+MONSTER: monster[1], random, (11,06)
+MONSTER: monster[2], random, (11,08)
+MONSTER: monster[2], random, (11,09)
+
+#  First hall:
+REGION: (16,07, 20,07), unlit, "ordinary"
+
+#  Throne room:
+REGION: (22,05, 28,09), lit, "throne"
+DOOR:  locked, (15, 07)
+DOOR:  locked, (21, 07)
+DOOR:  broken, (25, 10)
+#  Can't afford to repair it.
+
+#  Guard rooms:
+DOOR   :  locked    , (21, 05)
+MONSTER: 'o', random, (16, 05), asleep
+MONSTER: 'o', random, (17, 05), asleep
+MONSTER: 'o', random, (18, 05), asleep
+MONSTER: 'o', random, (19, 05), asleep
+
+DOOR   : closed        , (21, 09)
+MONSTER: random, random, (16, 09), asleep
+MONSTER: random, random, (17, 09), asleep
+MONSTER: random, random, (18, 09), asleep
+MONSTER: random, random, (19, 09), asleep
+
+#  Second hall:
+REGION: (30,07, 42,07), unlit, "ordinary"
+
+#  Treasure rooms:
+DOOR:  closed, (36, 06)
+DOOR:  closed, (36, 08)
+MONSTER: ':', "lizard", (36,05)
+MONSTER: ':', "lizard", (36,09)
+#  Can't afford dragons.
+
+DOOR:  locked, (37, 05)
+OBJECT: object[0], random, (39, 05)
+OBJECT: object[0], random, (40, 05)
+OBJECT: object[0], random, (41, 05)
+OBJECT: object[0], random, (42, 05)
+
+DOOR:  locked, (35, 09)
+OBJECT: random, random, (30, 09)
+OBJECT: random, random, (31, 09)
+OBJECT: random, random, (32, 09)
+OBJECT: random, random, (33, 09)
+
+DOOR:  locked, (37, 09)
+OBJECT: object[1], random, (39, 09)
+OBJECT: object[1], random, (40, 09)
+OBJECT: object[1], random, (41, 09)
+OBJECT: object[2], random, (42, 09)
+
+DOOR:  closed, (35, 05)
+OBJECT: '%', "egg" , (30, 05)
+OBJECT: '%', "egg" , (30, 05)
+OBJECT: '%', "egg" , (30, 05)
+#  This space intentionally left blank.
+OBJECT: '%', random, (32, 05)
+OBJECT: '%', "egg" , (32, 05)
+OBJECT: '%', "egg" , (32, 05)
+OBJECT: '%', "egg" , (32, 05)
+OBJECT: '%', random, (33, 05)
+OBJECT: '%', "egg" , (33, 05)
+OBJECT: '%', "egg" , (33, 05)
+OBJECT: '%', "egg" , (33, 05)
+#  Author's signature, right at the missing eggs:
+OBJECT: '`', "statue", (31, 05), "dwarf", 0
+#  In practice, many of the eggs hatch, so it's hard to see the
+#  implication:  stoned by eating a cockatrice egg.
+
+OBJECT   : '(', "large box", place[0]
+ENGRAVING: place[0], dust, "Owlbreath"
+
+ENGRAVING: (44,07), dust, "No milk today:  under siege"
+
+#  Level builder's signature:
+ENGRAVING: (09,03), engrave, "This Fortress of Ultimate Evil was built by a company that wishes to remain anonymous."
+
+WALLIFY
+
+
+
+#  "Foretaste of Gehennom", by Rob Ellwood
+
+MAZE : "rnd1-2", ' '
+FLAGS: shortsighted
+
+MESSAGE: "The air here is cold and dank."
+MESSAGE: "You see creatures huddling in a circle of flickering light."
+
+GEOMETRY: center, center
+
+#          10        20        30        40        50        60        70
+#012345678901234567890123456789012345678901234567890123456789012345678901234
+MAP
+0..........................PPPPPPPPPPPPPPPP...................PPPPPPPPP.....
+1............................PPPPPPPPPP........................PPPPPPPPPPP..
+2..............III................................................PPPPPPPPP.
+3............IIIIPPP...........................BBBBBB.................PPPPPP
+4...........IPPP{PPPPP........................BB....BB....................PP
+5..BBBBBB...PPPPPPPPP........................BB......BB.....................
+6.BB....BB...PPPPPP..........................B........B.....................
+7BB......BB....PP............................BBB.....BB.....................
+8BB......BB...............BBBB.................BB...BB......................
+9.BB....BB..............BBB..BB..III............BBBBB.......................
+0..BBBBBB..............BB....BBIIIIPPP......................................
+1......................BB....BIIPPPPPPP............IIII.....................
+2.......................BB..BBIPPPPPPPP.........IIIIIPPPPPI.................
+3..........IIIII.........B.BB..PPPPPPP.........IIIIPPP.PPPPPP...............
+4.......IIIIIPPPPP......BB.B.....PPP..........IIPPPPP{PPPPPPPP..............
+5.....IIIIPPPPPPP.....BBB..BBB....I...........IPPPPPPPPPPPPPPP..............
+6.....IPPPPPPPP......BB......BB...II...........PPPPPPPPPPPPPP...............
+7......PPPPP.........B........B....IIII..........PPPPPPPPP..................
+8........PP..........BBB.....BB....IIPPPPI..........PPP.....................
+9.........P............BBB..BB.....IPPPPPPPPI...............................
+0........................BBBB....IIPPPPPPPPPPPPPPI..........................
+ENDMAP
+#012345678901234567890123456789012345678901234567890123456789012345678901234
+#          10        20        30        40        50        60        70
+
+RANDOM_MONSTERS: 'o', 'o', 'o', 'o', 'o', 'o', 'd'
+
+REGION: (00,00, 74,20), unlit, "ordinary"
+
+STAIR: (04,09), up
+STAIR: (50,00, 74,20), (0,0, 0,0), down
+
+TRAP:  "fire", (04,07)
+REGION: (03,06, 04,07), lit, "ordinary", filled, true
+
+TRAP:  "fire", (24,17)
+TRAP:  "fire", (26,11)
+REGION: (24,17, 25,18), lit, "ordinary", filled, true
+REGION: (29,11, 29,12), lit, "ordinary"
+
+TRAP:  "fire", (49,06)
+REGION: (49,06, 50,07), lit, "ordinary", filled, true
+MONSTER      : monster[0], random, (48,05)
+MONSTER [50%]: monster[0], random, (49,05), asleep
+MONSTER [50%]: monster[0], random, (49,07)
+MONSTER [50%]: monster[0], random, (50,06)
+MONSTER [50%]: monster[1], random, (48,06), asleep
+MONSTER [50%]: monster[2], random, (48,05)
+
+MONSTER [50%]:' ',"ghost",random, peaceful
+MONSTER [50%]:' ',"ghost",random, peaceful
+MONSTER [50%]:' ',"ghost",random, peaceful
+MONSTER [50%]:' ',"ghost",random, peaceful
+MONSTER [50%]:' ',"ghost",random, peaceful
+MONSTER [50%]:' ',"ghost",random, peaceful
+MONSTER      :' ',"ghost",random, peaceful
+MONSTER      :' ',"ghost",random, peaceful
+
+TRAP:  "fire", (53,13)
+REGION: (51,11, 55,15), lit, "ordinary"
+REGION: (52,10, 54,16), lit, "ordinary"
+REGION: (50,12, 56,14), lit, "ordinary"
+
+TRAP[50%]: random, random
+TRAP[50%]: random, random
+TRAP[50%]: random, random
+TRAP[50%]: random, random
+TRAP[50%]: random, random
+TRAP[50%]: random, random
+TRAP[50%]: random, random
+TRAP[50%]: random, random
+
+OBJECT[20%]: random, random, random
+OBJECT[20%]: random, random, random
+OBJECT[20%]: '$'   , random, random
+
+MONSTER      : ';', "jellyfish", (34,10)
+MONSTER [50%]: ';', "jellyfish", (34,12)
+MONSTER [50%]: ';', "jellyfish", (50,14)
+MONSTER [50%]: ';', "jellyfish", (32,00)
+MONSTER [50%]: ';', "jellyfish", (66,02)
+MONSTER [50%]: ';', "jellyfish", (40,20)
+MONSTER [50%]: ';', "jellyfish", (10,16)
+
+OBJECT [60%]: '`', "boulder", random
+OBJECT [60%]: '`', "boulder", random
+OBJECT [60%]: '`', "boulder", random
+OBJECT [60%]: '`', "boulder", random
+OBJECT [60%]: '`', "boulder", random
+
+
+#  Level builder's signature:
+ENGRAVING:  (00,00), burn, "This foretaste of Gehennom was brought to you by Orcus.  Enjoy!"
+
+#  Author's signature
+OBJECT:  '%',"corpse", (02, 01), "dwarf", 0
+
+
+
+
+
+#  "The Wizard's Park", by Rob Ellwood
+
+#  Many of the items are cursed:  tainted by Moloch.  And those T-shirts
+#  are made out of incredibly shoddy material.
+#
+#  It's possible for the staircases to Sokoban or the Mines to start
+#  in the store.  Good.  About 1% of the time, they start on the island,
+#  making access difficult.  I didn't make the island off-limits for 
+#  staircases.  Instead, at one spot, you can dive in and (sometimes)
+#  come out on the far shore.  Challenges *add* to the game.
+
+MAZE: "rnd1-3", ' '
+
+FLAGS: shortsighted, arboreal
+
+MESSAGE: "A voice speaks from out of the air:"
+MESSAGE: " 'This park is for loyal minions of the Wizard of Yendor."
+MESSAGE: "  Intruders will be sacrificed to Moloch.'"
+
+GEOMETRY: center, center
+
+#          10        20        30        40        50        60        70
+#012345678901234567890123456789012345678901234567890123456789012345678901234
+MAP
+0---------------------------------------------------------------------------
+1|.......F...........................................PPPPPPPPPPPPP.........|
+2|.......+........T.....................................PPPPPPPPPPP........|
+3|.......F.......TTT.........................................PPPPPPPP......|
+4|--------........TTT........................{..................PPPPPPP....|
+5|..............T...T........T................P....................PPPPP...|
+6|..TT.......................TT...............PPP...................PPPPP..|
+7|...TTT....................TTT..............PPPPPP..................PPPP..|
+8|....TT...................TTT...............PPPPPPPP.................PPP..|
+9|.....TT......TTT.........T..................PPPPPPPPPPP..............PPP.|
+0|...........TTTT......TT.......................PPPPPPPPPPPP............PP.|
+1|..........TT..........TTTT.........................PPPPPPPPPPPPP.....PPPP|
+2|.........................TTT..............................PPPPPPPPPPPPPPP|
+3|T....TT........T...........T...........................................PP|
+4|TTTTTTTT|.....TTT.......................................................P|
+5|BBBBBBBB|......TTT..........T...................|----|...................|
+6|........+........T..........T...................|....|..........P{P......|
+7|..|-+|..|FFFFFFFFFFFFFFF...TT...................+....|.........PPPPP.....|
+8|..|..|..|..|..|..|..|..F.TTT....................|----|..........PPPPP....|
+9|..|..|..|..|..|..|..|..F.T.........................................PPP...|
+0---------------------------------------------------------------------------
+ENDMAP
+#012345678901234567890123456789012345678901234567890123456789012345678901234
+#          10        20        30        40        50        60        70
+
+
+RANDOM_MONSTERS: 'o', 'O', 'g', 'h', 'l', 'm', 'C'
+
+REGION: (00,00, 74,20), lit, "ordinary", filled
+
+
+#  Entrance room:
+
+STAIR: (03,01), up
+STAIR: (03,03), down
+
+DOOR:  open, (08, 02)
+ENGRAVING:  (07,02), engrave, "A holy day for Moloch:  free admission."
+
+#  Level builder's signature:
+ENGRAVING:  (01,01), engrave, "Entry is restricted to minions of the third rank and above."
+
+
+#  Zoo:
+MONSTER [90%]: 'f', "tiger"    , (11,19)
+MONSTER [90%]: 'q', "mumak"    , (14,18)
+MONSTER [90%]: 'd', "wolf"     , (17,18), hostile, "Lassier"
+MONSTER [90%]: 'Y', "ape"      , (20,18)
+MONSTER [90%]: ':', "crocodile", (23,18)
+ENGRAVING    : (17,16), engrave, "Do not pet wolf with remaining arm."
+OBJECT  [50%]: '%', "huge chunk of meat", (11,19), cursed, 0
+OBJECT  [50%]: '%', "huge chunk of meat", (17,19), cursed, 0
+OBJECT  [50%]: '%', "huge chunk of meat", (23,19), cursed, 0
+
+
+#  Food stand:
+REGION: (50,16, 53,17), lit, "food shop"
+DOOR:  open, (49, 17)
+
+OBJECT[70%]: '%', "candy bar" , (51,16),   cursed, 0
+OBJECT[20%]: '%', "candy bar" , (51,16)
+OBJECT[70%]: '%', "candy bar" , (51,17),   cursed, 0
+OBJECT[20%]: '%', "candy bar" , (51,17)
+OBJECT[45%]: '%', "meat stick", (52,16),   cursed,  0, "beef jerky"
+OBJECT[15%]: '%', "meat stick", (52,16), uncursed,  0, "beef jerky"
+OBJECT[45%]: '%', "meat stick", (52,17),   cursed,  0, "beef jerky"
+OBJECT[15%]: '%', "meat stick", (52,17), uncursed,  0, "beef jerky"
+OBJECT[20%]: '[', "T-shirt"   , (53,16),   cursed, -1, "The Wizard's Park"
+OBJECT[10%]: '[', "T-shirt"   , (53,16), uncursed, -1, "The Wizard's Park"
+OBJECT[20%]: '[', "T-shirt"   , (53,17),   cursed, -1, "I saw Bruce at The Wizard's Park"
+OBJECT[ 7%]: '[', "T-shirt"   , (53,17), uncursed, -1, "I saw Bruce at The Wizard's Park"
+
+#  Employees:
+MONSTER: '@', "watchman"     , random  , peaceful, "Groundskeeper Ed"
+MONSTER: '@', "watchman"     , random  , peaceful
+MONSTER: '@', "watchman"     , (06, 03), peaceful
+MONSTER: '@', "watch captain", random  , peaceful
+MONSTER: ';', "jellyfish"    , (50,09) , hostile
+MONSTER: ';', "shark"        , (61,01) , peaceful, "Bruce"
+#   "Bruce" was the robot shark used in Jaws.
+CONTAINER: '`', "statue", (49,03), "shark", 0, "Bruce"
+ENGRAVING: (49,03), burn, "See Bruce eat Moloch's enemies alive!"
+
+
+
+#  Tool shed:
+DOOR     : locked , (09, 16)
+ENGRAVING: (10,16), engrave , "Employees only."
+REGION   : (01,16 ,  08, 19), unlit, "ordinary", unfilled, true
+REGION   : (04,18 ,  05, 19), unlit, "ordinary", unfilled
+DOOR     : locked , (05, 17)
+
+#  People know that random pick-axes are uncursed:  OK.  For cans
+#  of grease, the last number on the line is the number of charges.
+OBJECT[50%]: '(', "pick-axe"      , (04,18)
+OBJECT[35%]: ')', "axe"           , (04,19), cursed, 0
+OBJECT[15%]: ')', "axe"           , (04,19)
+OBJECT[40%]: ')', "war hammer"    , (05,19), cursed, 0
+OBJECT[10%]: ')', "war hammer"    , (05,19)
+OBJECT[10%]: '(', "can of grease" , (04,18), cursed, 2
+OBJECT[10%]: '(', "can of grease" , (04,18), cursed, 5
+OBJECT[10%]: '(', "can of grease" , (04,18), cursed, 8
+OBJECT[10%]: '(', "can of grease" , (04,18)
+OBJECT[10%]: '(', random          , (04,18)
+#  "Pruning hook":
+OBJECT     : ')', "guisarme"      , (05,18), cursed, 0
+OBJECT[40%]: ')', "guisarme"      , (05,18)
+OBJECT[30%]: '[', "leather gloves", (04,19), cursed, 0
+OBJECT[10%]: '[', "leather gloves", (04,19)
+OBJECT[45%]: '[', "high boots"    , (04,18), cursed, 0
+OBJECT[15%]: '[', "high boots"    , (04,18)
+
+
+#  Peaceful minions:
+MONSTER [50%]: 'o'       , random , random, peaceful
+MONSTER [50%]: monster[0], random , random, peaceful
+MONSTER [50%]: monster[0], random , random, peaceful
+MONSTER [50%]: monster[1], random , random, peaceful
+MONSTER [50%]: monster[2], random , random, peaceful
+MONSTER [90%]: 'B'       , "raven", random, peaceful
+MONSTER [30%]: 'B'       , "raven", random, peaceful
+
+
+#  Decoration:
+CONTAINER: '`', "statue", (71,02), "titan", 0, "The Wizard of Yendor"
+OBJECT   : '"', "cheap plastic imitation of the Amulet of Yendor", contained, cursed, 0
+ENGRAVING: (71,02), engrave, "Our Glorious Leader"
+OBJECT[30%]: '%', "candy bar" , random, cursed, 0
+OBJECT[30%]: '%', "candy bar" , random, cursed, 0
+OBJECT[10%]: '%', "candy bar" , random
+OBJECT[10%]: '%', "candy bar" , random
+OBJECT[30%]: '?', "blank paper", random, cursed, 0, "A list of The Park's attractions"
+OBJECT[10%]: '?', "blank paper", random, cursed, 0, "A list of The Park's attractions"
+
+
+#  Author's signature:
+OBJECT[50%]: '`', "statue", random, "dwarf", 0, "Colonel Rob"
+
+#  And, to finish, a long section for the gravel pathway.  
+#  Only 128 OBJECTs per level, or I would put several
+#  flintstones per square.  The path is at the B's:
+#|.......F...........................................
+#|.......+BB......T.......BBBB..........................
+#|.......F..B....TTT.....B....BBB.......................
+#|--------...B....TTT...B........B.......................
+#|...........B..T...T...B....T....B........................
+#|..TT........B........B.....TT....B..........PPP
+#|...TTT.......BB....BB.....TTT.....B........PPPPPP
+#|....TT.........BBBB......TTT......B........PPPPPPPP
+#|.....TT......TTT.........T.........B........PPPPPPPPPPP
+#|...........TTTT......TT............B..........PPPPPPPPPPP
+#|..........TT..........TTTT.........B...............PPPPPPP
+#|.........................TTT........B.....................
+#|T....TT........T...........T........B.....................
+#|TTTTTTTT......TTT....................B....................
+#|........|......TTT..........T.........B.........|----|....
+#|..----..|........T..........T..........B........|....|....
+#|..+..|..|..................TT...........BB......+....|....
+#|..|..|..|................TTT..............BBBB..|----|....
+#|..----..|................T................................
+
+OBJECT:'*',"flint",(09,02), cursed, 0
+OBJECT:'*',"flint",(10,02), cursed, 0
+OBJECT:'*',"flint",(11,03), cursed, 0
+OBJECT:'*',"flint",(12,04), cursed, 0
+OBJECT:'*',"flint",(12,05), cursed, 0
+OBJECT:'*',"flint",(13,06), cursed, 0
+OBJECT:'*',"flint",(14,07), cursed, 0
+OBJECT:'*',"flint",(15,07), cursed, 0
+OBJECT:'*',"flint",(16,08), cursed, 0
+OBJECT:'*',"flint",(17,08), cursed, 0
+OBJECT:'*',"flint",(18,08), cursed, 0
+OBJECT:'*',"flint",(19,08), cursed, 0
+OBJECT:'*',"flint",(20,07), cursed, 0
+OBJECT:'*',"flint",(21,07), cursed, 0
+OBJECT:'*',"flint",(22,06), cursed, 0
+OBJECT:'*',"flint",(23,05), cursed, 0
+OBJECT:'*',"flint",(23,04), cursed, 0
+OBJECT:'*',"flint",(24,03), cursed, 0
+OBJECT:'*',"flint",(25,02), cursed, 0
+OBJECT:'*',"flint",(26,02), cursed, 0
+OBJECT:'*',"flint",(27,02), cursed, 0
+OBJECT:'*',"flint",(28,02), cursed, 0
+OBJECT:'*',"flint",(29,03), cursed, 0
+OBJECT:'*',"flint",(30,03), cursed, 0
+OBJECT:'*',"flint",(31,03), cursed, 0
+OBJECT:'*',"flint",(32,04), cursed, 0
+OBJECT:'*',"flint",(33,05), cursed, 0
+OBJECT:'*',"flint",(34,06), cursed, 0
+OBJECT:'*',"flint",(35,07), cursed, 0
+OBJECT:'*',"flint",(35,08), cursed, 0
+OBJECT:'*',"flint",(36,09), cursed, 0
+OBJECT:'*',"flint",(36,10), cursed, 0
+OBJECT:'*',"flint",(36,11), cursed, 0
+OBJECT:'*',"flint",(37,12), cursed, 0
+OBJECT:'*',"flint",(37,13), cursed, 0
+OBJECT:'*',"flint",(38,14), cursed, 0
+OBJECT:'*',"flint",(39,15), cursed, 0
+OBJECT:'*',"flint",(40,16), cursed, 0
+OBJECT:'*',"flint",(41,17), cursed, 0
+OBJECT:'*',"flint",(42,17), cursed, 0
+OBJECT:'*',"flint",(43,18), cursed, 0
+OBJECT:'*',"flint",(44,18), cursed, 0
+OBJECT:'*',"flint",(45,18), cursed, 0
+OBJECT:'*',"flint",(46,18), cursed, 0
+
+
+
+#  "Broken Maze", by Rob Ellwood
+
+#  After applying MAZEWALK, the MAP gives a Gehennom-style maze with
+#  *lot* of holes in the walls.  I need a dummy line 0 to make MAZEWALK
+#  work correctly.  Do not WALLIFY, or MAZEWALK fails.
+#
+#  The right side of the map comes out funny, which I figure is a plus.
+
+
+MAZE: "rnd1-4", ' '
+
+GEOMETRY:center,center
+
+#          10        20        30        40        50        60        70
+#0123456789012345678901234567890123456789012345678901234567890123456789012
+MAP
+0----|                                                                -----
+1-...|-----------------------------------------------------------------...-
+2-...+. . - - - - - - - . . . . - - - - - . - - . - . . - . . - - - - +...-
+3-...|.-...-.--..-.-.-.-----...-...-----.------.---.-...--.---..-.{.---...-
+4-----. . - - - - - . - - . - - - . - - . - - - - . . - - . - - - - . -----
+5-------.-----------.-----...-.--------...---.--------.---....-------------
+6-- . - - - - - . . - . - - - - - - . . . - - - . - - . - . . . - - - . . -
+7----..-.--.--...-.-..---.---....--.---.--.---..---..-.--....-.-..---.-.---
+8-- - - . - - . - ------------. - - - - - - - - . - . - . - - . - . - . - -
+9---.--..--.-..---.-........---.-...-----....----.--.--.--.-.--...---..-..-
+0-- - . . - - - - --........+ - . - - - - . - . - - - - - - . - - - - . . -
+1--...--....--------........--..----..---..-.--.------.----.----...---.----
+2-- - - - - - - . .------------ - - - - - - . - . . - . . - . - - - - . - -
+3--.-...--.-.----.--....-----.-------..-----.--....---.-.------.--.-----..-
+4-- - . . - - . - . - - . - . . - - . . . - - . - . - - - - . - - . . . . -
+5----|-------....----..--...-.-.---.-.-.---.-.------------..--.-..----|----
+6-...|- - . - - - - . - . - - . - . - . - - - - - - . - . . - - . . . |...-
+7-...+..--.--.---.-.-----...-.-.--..--.--.-.-------..-----------..-.-.+...-
+8-...|- . - - . - - . - - - - - - . . . - - - - - - - - - . - - - - - |...-
+9--------------------------------------------------------------------------
+ENDMAP
+#012345678901234567890123456789012345678901234567890123456789012345678901234
+#          10        20        30        40        50        60        70
+
+RANDOM_PLACES : (02,02), (02,17), (71,02), (71,17)
+
+REGION: (00,00, 73,19), lit, "ordinary", unfilled
+
+DOOR:  open, (27, 10)
+MAZEWALK:    (27, 10), east
+
+STAIR: (23,09), up
+
+#  Bug:  a randomly placed staircase is sometimes "eaten" by mazewalk.
+#  The staircase still works, but it's shown as plain floor on the map.
+#  Workaround:  add four corner rooms; put the stairs down in a random room.
+
+DOOR:  closed, (04, 02)
+DOOR:  closed, (04, 17)
+DOOR:  closed, (69, 02)
+DOOR:  closed, (69, 17)
+STAIR: place[0], down
+
+
+TRAP[50%]: random, random
+TRAP[50%]: random, random
+TRAP[50%]: random, random
+TRAP[50%]: random, random
+
+MONSTER[50%]: random, random, random
+MONSTER[50%]: random, random, random
+MONSTER[50%]: random, random, random
+MONSTER[50%]: random, random, random
+MONSTER[50%]: random, random, random
+
+OBJECT[50%]: random, random, random
+OBJECT[50%]: random, random, random
+OBJECT[50%]: random, random, random
+OBJECT[50%]: random, random, random
+OBJECT[50%]: random, random, random
+
+#  Only put boulders where they can be pushed:  unpushable boulders
+#  on top of staircases would be bad.
+OBJECT[60%]: '`', "boulder", (08,03)
+OBJECT[60%]: '`', "boulder", (57,06)
+OBJECT[60%]: '`', "boulder", (25,17)
+OBJECT[60%]: '`', "boulder", (69,09)
+
+OBJECT[50%]: '$', random, random
+OBJECT[50%]: '$', random, random
+OBJECT[50%]: '$', random, random
+OBJECT[50%]: '$', random, random
+OBJECT[50%]: '$', random, random
+
+OBJECT[30%]: '(', "chest", random
+
+REGION: (51,11, 55,15), unlit, "ordinary"
+REGION: (52,10, 54,16), unlit, "ordinary"
+REGION: (50,12, 56,14), unlit, "ordinary"
+
+REGION: (21,07, 25,11), unlit, "ordinary"
+REGION: (22,06, 24,12), unlit, "ordinary"
+REGION: (20,08, 26,10), unlit, "ordinary"
+
+
+#  Level builder's signature:
+ENGRAVING:(01,01), engrave, "Dwarves screwed up; failed the minotaur's QA.  Sell to WoY; he'll buy anything."
+#  QA = Quality Assurance
+
+#  Author's signature:
+OBJECT[50%]: '`', "statue", random, "dwarf", 0, "Colonel Rob"
+
+
+
+#  "Citadel", by Rob Ellwood
+
+MAZE: "rnd1-5", ' '
+
+#  This level is designed to look unusual:  lit corridors, no doors,
+#  a central area, partly symmetric, improperly walled.
+
+#  The message assumes that the player can see.  And the orcs will
+#  be goblins for low-level characters.  Oh, well.
+
+MESSAGE: "There are half-eaten human corpses here."
+MESSAGE: "The feasting slowly stops as you are noticed."
+
+GEOMETRY:center,center
+
+#          10        20        30        40        50        60        70
+#012345678901234567890123456789012345678901234567890123456789012345678901234
+MAP
+0......|                                                             |......
+1......|                                                             |......
+2......|---------------------                    --------------------|......
+3...........................|--------------------|..........................
+4---|.|-------------------|........................|-----------|.|----------
+5   |..............|      |........................|           |.|          
+6   -|.............|      |.......|--------|.......|           |.|          
+7    |.............|      |.....--|........|--.....|        ---|.|--        
+8    ---|.|---------      |.....|............|.....|        |......|        
+9       |.|               |.....|.....{......+.....|        |......|        
+0 ----- |.|               |.....|............|.....|        |......|        
+1 |...| |.| -----------   |.....|............|.....|        |......|        
+2 |.......| |.........|---|.....---........---.....|        ---|.|-------   
+3 |...|     |.............S.......|--------|.......|           |.|-|....|---
+4 |...|     |.........|---|........................|           |...........|
+5 |...|     -----------   |........................|           ----|....-|.|
+6--|.|--------------------|.|--------------------|.|---------------------|.|
+7...........................|                    |..........................
+8......|---------------------                    --------------------|......
+9......|                                                             |......
+0......|                                                             |......
+ENDMAP
+#012345678901234567890123456789012345678901234567890123456789012345678901234
+#          10        20        30        40        50        60        70
+
+RANDOM_PLACES:  (38,04), (38,04), (29,10), (29,10), (27,07), (27,07)
+RANDOM_MONSTERS: 'o', 'o', 'o', 'o', 'o', 'o', 'o', 'd'
+
+
+REGION: (00,00, 74,20), lit, "ordinary", unfilled
+REGION: (11,17, 16,18), unlit, "ordinary"
+REGION: (60,08, 65,11), unlit, "ordinary"
+
+#  Enter the level near the orcs, so the initial message makes sense:
+TELEPORT_REGION:(26,04, 30,15), (0,0, 0,0), down
+TELEPORT_REGION:(26,04, 30,15), (0,0, 0,0), up
+
+STAIR: (27, 05), up
+STAIR: (27, 14), down
+
+OBJECT:  '%',"corpse", (38 ,04), "soldier", 0
+MONSTER: 'o', random , (38, 04)
+MONSTER: 'o', random , (39, 04), asleep
+
+OBJECT:  '%',"corpse", (29 ,10), "soldier", 0
+MONSTER: 'o', random , (29, 10)
+MONSTER: 'o', random , (28, 14)
+
+OBJECT:  '%',"corpse", (27 ,07), "orc", 0
+
+#  Bits of loot at the corpses:
+OBJECT[50%]: '[', random          , place[0]
+OBJECT[40%]: '[', random          , place[1]
+OBJECT[70%]: ')', random          , place[2]
+OBJECT[50%]: ')', random          , place[3]
+OBJECT[10%]: '[', "leather gloves", place[4]
+
+#  Random gang of orcs in the citadel.  Well, usually orcs.
+MONSTER: monster[0], random    , (38, 09)
+MONSTER: monster[0], random    , (39, 09)
+MONSTER: monster[0], random    , (37, 10)
+MONSTER: 'o'       , "Uruk-hai", (37, 09)
+
+#  Lock the door, so the player controls the battle.  Pin the 'o' in
+#  place (and weaken them a bit) with a trap just inside the door:
+DOOR     : locked, (44, 09)
+TRAP[50%]: random, (43, 09)
+
+#  Approximate the loot on a normal level:
+OBJECT [30%]: '(', "large box", random
+
+OBJECT [50%]:  random, random, random
+OBJECT [50%]:  random, random, random
+OBJECT [50%]:  random, random, random
+OBJECT [50%]:  random, random, random
+OBJECT [50%]:  random, random, random
+
+OBJECT [50%]:  '$', random, random
+OBJECT [50%]:  '$', random, random
+OBJECT [50%]:  '$', random, random
+OBJECT [50%]:  '$', random, random
+
+OBJECT [60%]: '`', "boulder", random
+
+#  Author's signature; level builder's signature:
+ENGRAVING:  (0,0), mark, "This area provided courtesy of CROGGLED"
+#  Colonel Rob's Global Gang of Level Designers.
+
+WALLIFY
+
+
+
+#  "Suburb", by Rob Ellwood
+
+MAZE: "rnd1-6", ' '
+
+#  The following oddities are deliberate:  rock pillars, a missing
+#  bit of wall for one room, rock without surrounding walls (don't
+#  use WALLIFY on this level), upper left corner bricked up.
+
+GEOMETRY:center,center
+
+#          10        20        30        40        50        60        70
+#012345678901234567890123456789012345678901234567890123456789012345678901234
+MAP
+0---------------------------------------------------------------------------
+1 |..|......|...|......|....................|......|...|......|............|
+2 |..|......+...+......|....................|......+...+......|....----....|
+3 |..|......|...|......|.......|--------|...|......|...|......|----|..|----|
+4-|..|------|...|------|.......|........|...|------|...|------|....|..|....|
+5|.............................|........|..........................|..|....|
+6|... .T........... ...........|----+---|..........................+..|....|
+7|... .....T....... ..........................-----------..........|..|....|
+8|... ...T...{....................|--|........|......|..|..........|..|....|
+9|... .............|-------|......|  |........|......|..|..........|..|....|
+0|.................|.......|......|  |........|......|..+....|--|..|..|....|
+1|..-----..........|.......+......|--|........+......|..|....|  |..|--|....|
+2|..-   -.        .|.......|..................|......|--|....|  |..........|
+3|..-   -..........|-------------.............|.........|....|--|..........|
+4|..-----..............|........|.............|---------|..................|
+5|..-...-..............+........|..........................................|
+6|..-...----+----......|........|........|------....|------|...............|
+7|..-...........-......|---------........|......|...|......|...........----|
+8|..-...........-........................|......+...+......|...........+...|
+9|..-...........-........................|......|...|......|...........|.K.|
+0---------------------------------------------------------------------------
+ENDMAP
+#012345678901234567890123456789012345678901234567890123456789012345678901234
+#          10        20        30        40        50        60        70
+
+DOOR: locked, (11,02)
+DOOR: random, (50,02)
+DOOR: random, (54,02)
+DOOR: random, (35,06)
+DOOR: random, (45,11)
+DOOR: random, (55,10)
+DOOR: random, (66,06) 
+DOOR: random, (22,15)
+DOOR: random, (11,16)
+DOOR: random, (47,18)
+DOOR: random, (51,18)
+
+REGION: (00,00, 74,20), lit, "ordinary", unfilled
+
+STAIR: random, up
+STAIR: random, down
+
+REGION:(19,10, 25,12), lit, "shop"
+DOOR: open  , (26,11)
+
+MONSTER:  '@', "watch captain", (18, 03), peaceful, asleep
+DOOR: closed, (15,02)
+
+#  Two mothers and a bunch of kids:
+MONSTER [80%]:  'k', "large kobold", (45,19), peaceful
+MONSTER [80%]:  'k', "large kobold", (55,19), peaceful
+MONSTER      :  'k',       "kobold", random , peaceful
+MONSTER [90%]:  'k',       "kobold", random , peaceful
+MONSTER [70%]:  'k',       "kobold", random , peaceful
+MONSTER [50%]:  'k',       "kobold", random , peaceful
+MONSTER [50%]:  'k',       "kobold", random , peaceful
+
+
+FOUNTAIN:  (08, 07)
+
+MONSTER [50%]:  'd'   , random      , random, peaceful
+MONSTER [50%]:  'd'   , "little dog", random, peaceful
+MONSTER [50%]:  'f'   , "kitten"    , random, peaceful
+MONSTER [50%]:  'o'   , random      , random, peaceful
+MONSTER [40%]:  random, random      , random, peaceful
+MONSTER [20%]:  random, random      , random, peaceful
+MONSTER [30%]:  random, random      , random, hostile
+MONSTER [20%]:  random, random      , random, hostile
+
+#  Pay toilet:
+DOOR: closed, (70,18)
+ENGRAVING:  (69,18), burn, "Two zorkmids"
+TRAP  : "hole", (73, 19)
+
+#  Very little loot:
+OBJECT [30%]: '('    , "large box", random
+OBJECT [15%]:  random, random     , random
+OBJECT [05%]:  random, random     , random
+
+#  Level builder's signature:
+ENGRAVING:  (02,01), burn, "This rest stop courtesy of the Yendorian Department of Loways"
+
+#  Author's signature
+OBJECT:  '%',"corpse", (02, 01), "dwarf", 0
+ENGRAVING:  (69,18), dust, "Iron horse hit me; watch out fo-~~."
+
+
+
+
+#  This level is close to a normal level, except for the lack
+#  of special rooms.
+
+LEVEL: "rnd1-8"
+
+ROOM: "ordinary" , random, random, random, random
+STAIR: random, up
+
+ROOM: "ordinary" , random, random, random, random
+STAIR: random, down
+
+ROOM: "ordinary" , random, random, random, random
+TRAP: random, random
+MONSTER: random, random, random
+OBJECT: random, random, random
+
+ROOM: "ordinary" , random, random, random, random
+TRAP: random, random
+
+OBJECT: random, random, random
+GOLD: random, random
+
+ROOM: "ordinary" , random, random, random, random
+OBJECT: random, random, random
+GOLD: random, random
+
+ROOM: "ordinary" , random, random, random, random
+
+ROOM: "ordinary" , random, random, random, random
+MONSTER: random, random, random
+
+#  Author's signature.  RGRN => Regron.
+OBJECT:  '%',"corpse", (01,01), "dwarf", 0
+ENGRAVING:  (01,01), dust, "Regron ain't here."
+
+ROOM: "ordinary" , random, random, random, random
+GOLD: random, random
+
+ROOM: "ordinary" , random, random, random, random
+MONSTER: random, random, random
+OBJECT: '(', "large box", random
+
+RANDOM_CORRIDORS
+
+
+#  This level is close to a normal level, except for the lack
+#  of special rooms.
+
+LEVEL: "rnd1-9"
+
+ROOM: "ordinary" , random, random, random, random
+STAIR: random, up
+
+ROOM: "ordinary" , random, random, random, random
+STAIR: random, down
+
+ROOM: "ordinary" , random, random, random, random
+TRAP: random, random
+MONSTER: random, random, random
+OBJECT: random, random, random
+
+ROOM: "ordinary" , random, random, random, random
+TRAP: random, random
+
+OBJECT: random, random, random
+GOLD: random, random
+
+ROOM: "ordinary" , random, random, random, random
+OBJECT: random, random, random
+GOLD: random, random
+
+ROOM: "ordinary" , random, random, random, random
+
+ROOM: "ordinary" , random, random, random, random
+MONSTER: random, random, random
+
+#  Author's signature.  RGRN => Regron.
+OBJECT:  '%',"corpse", (01,01), "dwarf", 0
+ENGRAVING:  (01,01), dust, "Warn Regron:  the Tentacled Ones have corrupted RDE."
+
+ROOM: "ordinary" , random, random, random, random
+GOLD: random, random
+
+ROOM: "ordinary" , random, random, random, random
+MONSTER: random, random, random
+OBJECT: '(', "large box", random
+
+RANDOM_CORRIDORS
+
+
+#  This level is close to a normal level, except for the lack
+#  of special rooms.
+
+LEVEL: "rnd1-10"
+
+ROOM: "ordinary" , random, random, random, random
+STAIR: random, up
+
+ROOM: "ordinary" , random, random, random, random
+STAIR: random, down
+
+ROOM: "ordinary" , random, random, random, random
+TRAP: random, random
+MONSTER: random, random, random
+OBJECT: random, random, random
+
+ROOM: "ordinary" , random, random, random, random
+TRAP: random, random
+
+OBJECT: random, random, random
+GOLD: random, random
+
+ROOM: "ordinary" , random, random, random, random
+OBJECT: random, random, random
+GOLD: random, random
+
+ROOM: "ordinary" , random, random, random, random
+
+ROOM: "ordinary" , random, random, random, random
+MONSTER: random, random, random
+
+#  Author's signature.  RGRN => Regron.
+OBJECT:  '%',"corpse", (01,01), "dwarf", 0
+ENGRAVING:  (01,01), dust, "You are hereby promoted to Trusted Lieutenant"
+#  Evil Overlord rule:  promotions to Trusted Lieutenant only happen
+#  posthumously.
+
+ROOM: "ordinary" , random, random, random, random
+GOLD: random, random
+
+ROOM: "ordinary" , random, random, random, random
+MONSTER: random, random, random
+OBJECT: '(', "large box", random
+
+RANDOM_CORRIDORS
+
+
+
+#  This level is close to a normal level, except for the lack
+#  of special rooms.
+
+LEVEL: "rnd1-11"
+
+ROOM: "ordinary" , random, random, random, random
+STAIR: random, up
+
+ROOM: "ordinary" , random, random, random, random
+STAIR: random, down
+
+ROOM: "ordinary" , random, random, random, random
+TRAP: random, random
+MONSTER: random, random, random
+OBJECT: random, random, random
+
+ROOM: "ordinary" , random, random, random, random
+TRAP: random, random
+OBJECT: random, random, random
+GOLD: random, random
+
+ROOM: "ordinary" , random, random, random, random
+OBJECT: random, random, random
+GOLD: random, random
+
+ROOM: "ordinary" , random, random, random, random
+
+ROOM: "ordinary" , random, random, random, random
+MONSTER: random, random, random
+
+OBJECT:  '%',"corpse", (01,01), "dwarf", 0
+ENGRAVING:  (01,01), dust, "CROGGLED got me."
+
+ROOM: "ordinary" , random, random, random, random
+GOLD: random, random
+
+ROOM: "ordinary" , random, random, random, random
+MONSTER: random, random, random
+OBJECT: '(', "large box", random
+
+RANDOM_CORRIDORS
+
+
+
+#	SCCS Id: @(#)castle.des	3.4	1995/11/19
+#	Copyright (c) 1989 by Jean-Christophe Collet
+# NetHack may be freely redistributed.  See license for details.
+
+
+#	rnd1 is a collection of non-random levels.  I want them to
+#  break the monotony of the normal room-based levels.  Goal:  
+#  have enough of them that players rarely see any particular one.
+#
+#	They look unusual.  This is deliberate. 
+#
+#     The average rnd1 level will eventually match the average
+#  normal level is terms of special rooms, loot, and whatnot.
+#  That isn't true yet.
+
+#  	Distribution of stuff which matches the normal levels:
+#
+#  Random objects: 3.1  (1/3 rooms start with 1.25 objects.  I should
+#       add a bit for garter snakes & cave spiders, which get objects
+#       to hide under when they're created.)
+#  Boxes/chests per level:  30% a box/chest + 16% a box/chest, with
+#    an even split between boxes & chests.  (Or close enough)
+#    Chest contents:  50% random object, 5 times.
+#    Box contents:  50% random object, 3 times.
+#  Rooms:  I suspect that it's 10, but the program stops when it 
+#    has problems fitting in another.  Wiz-mode tests showed an 
+#    average of 7.66 rooms per level.
+#  Monsters:  2.5  (1/3 rooms start with monsters)
+#  Gold    :  2.5  (1/3 rooms start with gold)
+#  Traps   :  1.1  (1/7 rooms at about level 5)
+#                  x = 8 - (level_difficulty()/6);
+#                  if (x <= 1) x = 2;
+#                  while (!rn2(x)) mktrap(0,0,croom,(coord*)0);
+#  Fountains :  0.75.  (1/10th of rooms have a fountain.  Fake the
+#                  distribution with 50% fountain + 25% fountain.)
+#  Sinks     :  1 every 8 levels.  (1/60 per room)
+#  Statues   :  1 every 3 levels.  (1/20 per room)
+#  Altar     :  1 every 8 levels.  (1/60 per room)
+#  Shrines   :  are these done separately?
+#  Graveyards:  None here; DL 12+ only.   
+#                  x = 80 - (depth(&u.uz) * 2);if (x < 2) x = 2;
+#                  if(!rn2(x)) mkgrave(croom);
+#  Vaults    :  1 every 2 levels.	
+#  Portal to Ludios:  1/3 per vault at DL 12+, BUT the odds are
+#                  probably zero for vaults on special levels).  
+
+#  Special rooms:  once you get one on a level, don't make any more.
+# 
+#  Shop:  3/DL, but you have to exclude levels where no rooms have
+#       only one door.  ~half are general, see room-331 for info.
+#  Throne room:  1/6, DL 5+
+#  Leprechaun hall 1/8, DL 6+
+#  Zoo 1/7, DL 7+
+#  Temple 1/5 DL 9+
+#  Beehive 1/5 DL 10+
+#  Graveyard 1/6 DL 12+  This rule disagrees with the code I have
+#      above.
+#  Anthole 1/8 DL 13+  food on 1/3 of squares.
+#  Barracks 1/4 DL 15+  chests or boxes on 1/20 of squares. 
+#  Swamp (1/6, DL 16+)
+#     Up to five rooms on the level become swamp rooms, with a chequered
+#     pattern of pools possibly containing giant eels, electric eels, or
+#     piranhas, alternating with dry land possibly containing a random
+#     fungus.
+#  Cockatrice nest 1/8, DL 17+, 
+#     statues of player-monsters on 1/3 of squares
+#     containing up to four random items.
+
+
+#  Signature blocks; use as required.
+
+#  Author's signature.  RGRN => Regron.
+#OBJECT:  '%',"corpse", (01,01), "dwarf", 0
+#ENGRAVING:  (01,01), dust, "Tell Regron I was Robbed"
+#ENGRAVING:  (01,01), dust, "Regron ain't here."
+#ENGRAVING:  (01,01), dust, "Warn Regron:  the Tentacled Ones have corrupted RDE."
+#ENGRAVING:  (01,01), dust, "You are hereby promoted to Trusted Lieutenant"
+#  Evil Overlord rule:  promotions to Trusted Lieutenant only happen
+#  posthumously.
+#ENGRAVING:  (01,01), dust, "Tell Regron I was Robbed"
+#ENGRAVING:  (01,01), dust, "CROGGLED got me."
diff -ruN nethack-3.4.0./sys/unix/Makefile.dat nethack-3.4.0/sys/unix/Makefile.dat
--- nethack-3.4.0./sys/unix/Makefile.dat	Thu Apr 11 17:58:28 2002
+++ nethack-3.4.0/sys/unix/Makefile.dat	Sun Jun 16 21:31:09 2002
@@ -85,7 +85,7 @@
 
 spec_levs: ../util/lev_comp \
 	bigroom.des castle.des endgame.des gehennom.des knox.des medusa.des \
-	mines.des oracle.des sokoban.des tower.des yendor.des
+	mines.des oracle.des sokoban.des tower.des yendor.des rnd1.des
 	../util/lev_comp bigroom.des
 	../util/lev_comp castle.des
 	../util/lev_comp endgame.des
@@ -97,6 +97,7 @@
 	../util/lev_comp sokoban.des
 	../util/lev_comp tower.des
 	../util/lev_comp yendor.des
+	../util/lev_comp rnd1.des
 	touch spec_levs
 
 quest_levs: ../util/lev_comp \
diff -ruN nethack-3.4.0./sys/unix/Makefile.top nethack-3.4.0/sys/unix/Makefile.top
--- nethack-3.4.0./sys/unix/Makefile.top	Sun Jun 16 21:30:50 2002
+++ nethack-3.4.0/sys/unix/Makefile.top	Sun Jun 16 21:31:09 2002
@@ -74,10 +74,11 @@
 DATHELP = help hh cmdhelp history opthelp wizhelp
 
 SPEC_LEVS = asmodeus.lev baalz.lev bigrm-?.lev castle.lev fakewiz?.lev \
-	juiblex.lev knox.lev medusa-?.lev minend-?.lev minefill.lev \
-	minetn-?.lev oracle.lev orcus.lev sanctum.lev soko?-?.lev \
+	juiblex.lev knox.lev medusa-?.lev m_end-?.lev minefill.lev \
+	m_twn-?.lev oracle.lev orcus.lev sanctum.lev soko?-?.lev \
 	tower?.lev valley.lev wizard?.lev \
-	astral.lev air.lev earth.lev fire.lev water.lev
+	astral.lev air.lev earth.lev fire.lev water.lev \
+	rnd1-?.lev
 QUEST_LEVS = ???-goal.lev ???-fil?.lev ???-loca.lev ???-strt.lev
 
 DATNODLB = $(VARDATND) license
