diff -ruN nethack-3.4.0.orig/include/you.h nethack-3.4.0/include/you.h
--- nethack-3.4.0.orig/include/you.h	Thu Apr 11 17:58:27 2002
+++ nethack-3.4.0/include/you.h	Sat Apr 13 16:30:42 2002
@@ -62,6 +62,7 @@
 	long	unvegan;	/* ... or any animal byproduct */
 	long	food;		/* ... or any comestible */
 	long	gnostic;	/* used prayer, priest, or altar */
+	long    role;           /* followed chivalry/bushido */
 	long	weaphit;	/* hit a monster with a weapon */
 	long	killer;		/* killed a monster yourself */
 	long	literate;	/* read something (other than BotD) */
diff -ruN nethack-3.4.0.orig/src/cmd.c nethack-3.4.0/src/cmd.c
--- nethack-3.4.0.orig/src/cmd.c	Sat Apr 13 01:11:53 2002
+++ nethack-3.4.0/src/cmd.c	Sat Apr 13 16:29:33 2002
@@ -1468,6 +1468,12 @@
 	if (!u.uconduct.gnostic)
 	    you_have_been("an atheist");
 
+	if ((Role_if(PM_KNIGHT) || Role_if(PM_SAMURAI)) && !u.uconduct.role) {
+            Sprintf(buf, "followed the code of %s",
+                    Role_if(PM_KNIGHT) ? "chivalry" : "bushido");
+            you_have_X(buf);
+        }
+	
 	if (!u.uconduct.weaphit)
 	    you_have_never("hit with a wielded weapon");
 #ifdef WIZARD
@@ -1557,6 +1563,12 @@
 	if (!u.uconduct.gnostic)
 	    dump("", "  You were an atheist");
 
+	if ((Role_if(PM_KNIGHT) || Role_if(PM_SAMURAI)) && !u.uconduct.role) {
+	    Sprintf(buf, "followed the code of %s",
+	        Role_if(PM_KNIGHT) ? "chivalry" : "bushido");
+	    dump("  You have ", buf);
+	}
+
 	if (!u.uconduct.weaphit)
 	    dump("", "  You never hit with a wielded weapon");
 #ifdef WIZARD
diff -ruN nethack-3.4.0.orig/src/dig.c nethack-3.4.0/src/dig.c
--- nethack-3.4.0.orig/src/dig.c	Thu Apr 11 17:58:27 2002
+++ nethack-3.4.0/src/dig.c	Sat Apr 13 16:19:52 2002
@@ -740,9 +740,11 @@
 	} else if (Role_if(PM_SAMURAI)) {
 	    adjalign(-sgn(u.ualign.type));
 	    You("disturb the honorable dead!");
+            u.uconduct.role++;
 	} else if ((u.ualign.type == A_LAWFUL) && (u.ualign.record > -10)) {
 	    adjalign(-sgn(u.ualign.type));
 	    You("have violated the sanctity of this grave!");
+            if (Role_if(PM_KNIGHT)) u.uconduct.role++;
 	}
 
 	switch (rn2(5)) {
diff -ruN nethack-3.4.0.orig/src/dokick.c nethack-3.4.0/src/dokick.c
--- nethack-3.4.0.orig/src/dokick.c	Thu Apr 11 17:58:27 2002
+++ nethack-3.4.0/src/dokick.c	Sat Apr 13 16:19:52 2002
@@ -68,6 +68,7 @@
 		(mon->mflee && !mon->mavenge))) {
 	    You_feel("like a caitiff!");
 	    adjalign(-1);
+            u.uconduct.role++;
 	}
 
 	/* squeeze some guilt feelings... */
diff -ruN nethack-3.4.0.orig/src/eat.c nethack-3.4.0/src/eat.c
--- nethack-3.4.0.orig/src/eat.c	Thu Apr 11 17:58:28 2002
+++ nethack-3.4.0/src/eat.c	Sat Apr 13 16:19:52 2002
@@ -231,6 +231,7 @@
 	} else if (Role_if(PM_KNIGHT) && u.ualign.type == A_LAWFUL) {
 			adjalign(-1);		/* gluttony is unchivalrous */
 			You_feel("like a glutton!");
+                        u.uconduct.role++;
 	}
 
 	exercise(A_CON, FALSE);
diff -ruN nethack-3.4.0.orig/src/shk.c nethack-3.4.0/src/shk.c
--- nethack-3.4.0.orig/src/shk.c	Thu Apr 11 17:58:28 2002
+++ nethack-3.4.0/src/shk.c	Sat Apr 13 16:25:24 2002
@@ -482,9 +482,11 @@
 	eshkp->robbed += total;
 	You("stole %ld %s worth of merchandise.",
 	    total, currency(total));
-	if (!Role_if(PM_ROGUE))	/* stealing is unlawful */
+	if (!Role_if(PM_ROGUE)) {
+	      /* stealing is unlawful */
 	    adjalign(-sgn(u.ualign.type));
-
+	    u.uconduct.role++;
+	  }
 	hot_pursuit(shkp);
 }
 
@@ -3291,6 +3293,7 @@
 	if (Role_if(PM_KNIGHT)) {
 	    You_feel("like a common thief.");
 	    adjalign(-sgn(u.ualign.type));
+            u.uconduct.role++;
 	}
 	return;
     }
@@ -3308,6 +3311,7 @@
 	if (Role_if(PM_KNIGHT)) {
 	    You_feel("like a common thief.");
 	    adjalign(-sgn(u.ualign.type));
+            u.uconduct.role++;
 	}
     } else if(!um_dist(shkp->mx, shkp->my, 5) &&
 		!shkp->msleeping && shkp->mcanmove &&
@@ -3534,6 +3538,7 @@
 		verbalize("Oh, yes!  You'll pay!");
 		hot_pursuit(shkp);
 		adjalign(-sgn(u.ualign.type));
+                u.uconduct.role++;
 	}
 }
 #endif /*OVLB*/
diff -ruN nethack-3.4.0.orig/src/uhitm.c nethack-3.4.0/src/uhitm.c
--- nethack-3.4.0.orig/src/uhitm.c	Thu Apr 11 17:58:28 2002
+++ nethack-3.4.0/src/uhitm.c	Sat Apr 13 16:19:52 2002
@@ -222,6 +222,7 @@
 	    u.ualign.record > -10) {
 	    You("caitiff!");
 	    adjalign(-1);
+            u.uconduct.role++;
 	}
 
 /*	attacking peaceful creatures is bad for the samurai's giri */
@@ -229,6 +230,7 @@
 	    u.ualign.record > -10) {
 	    You("dishonorably attack the innocent!");
 	    adjalign(-1);
+            u.uconduct.role++;
 	}
 
 /*	Adjust vs. (and possibly modify) monster state.		*/
@@ -862,9 +864,11 @@
 	    if Role_if(PM_SAMURAI) {
 		You("dishonorably use a poisoned weapon!");
 		adjalign(-sgn(u.ualign.type));
+                u.uconduct.role++;
 	    } else if ((u.ualign.type == A_LAWFUL) && (u.ualign.record > -10)) {
 		You_feel("like an evil coward for using a poisoned weapon.");
 		adjalign(-1);
+                if (Role_if(PM_KNIGHT)) u.uconduct.role++;
 	    }
 	    if (obj && !rn2(nopoison)) {
 		obj->opoisoned = FALSE;
diff -ruN nethack-3.4.0.orig/src/vault.c nethack-3.4.0/src/vault.c
--- nethack-3.4.0.orig/src/vault.c	Thu Apr 11 17:58:28 2002
+++ nethack-3.4.0/src/vault.c	Sat Apr 13 16:19:52 2002
@@ -264,6 +264,7 @@
 	    /* ignore trailing text, in case player includes character's rank */
 	    strncmpi(buf, plname, (int) strlen(plname)) != 0) {
 		adjalign(-1);		/* Liar! */
+                u.uconduct.role++;
 	}
 
 	if (!strcmpi(buf, "Croesus") || !strcmpi(buf, "Kroisos")
