diff -ruN nethack-3.4.0.orig/src/cmd.c nethack-3.4.0/src/cmd.c
--- nethack-3.4.0.orig/src/cmd.c	Thu Apr 11 17:58:27 2002
+++ nethack-3.4.0/src/cmd.c	Thu Apr 11 23:51:45 2002
@@ -463,7 +463,16 @@
 	    else aggravate();
 	} else if (Upolyd)
 		pline("Any special ability you may have is purely reflexive.");
-	else You("don't have a special ability in your normal form!");
+	else {
+	    static const char *tricks[] = {
+		"do a brief soft-shoe",
+		"juggle briefly",
+		"balance a pebble on your nose",
+		"do an impression of Elvis"
+	    };
+	    int i = rn2(SIZE(tricks));
+	    You("%s, but no one seems impressed.", tricks[i]);
+	}
 	return 0;
 }
 
