diff -urN nethack-3.3.1.orig/util/dlb_main.c nethack-3.3.1/util/dlb_main.c
--- nethack-3.3.1.orig/util/dlb_main.c	Wed May 10 02:18:02 2000
+++ nethack-3.3.1/util/dlb_main.c	Tue Mar 12 14:30:22 2002
@@ -53,7 +53,7 @@
 #define O_BINARY 0
 #endif
 
-#define MAX_DLB_FILES 200	/* max # of files we'll handle */
+#define MAX_DLB_FILES 500	/* max # of files we'll handle */
 #define DLB_VERS 1		/* version of dlb file we will write */
 
 /*
