diff -ruN nethack-3.4.0.orig/dat/opthelp nethack-3.4.0/dat/opthelp
--- nethack-3.4.0.orig/dat/opthelp	Thu Apr 11 17:58:27 2002
+++ nethack-3.4.0/dat/opthelp	Thu Apr 11 23:56:44 2002
@@ -56,6 +56,10 @@
 Boolean option if MFLOPPY was set at compile time:
 checkspace check free disk space before writing files to disk     [TRUE]
 
+Boolean option if SHOW_BORN was set at compile time:
+showborn   show numbers of created monsters after the number
+           vanquished                                             [FALSE]
+
 Boolean option if EXP_ON_BOTL was set at compile time:
 showexp    display your accumulated experience points             [FALSE]
 
diff -ruN nethack-3.4.0.orig/dat/wizhelp nethack-3.4.0/dat/wizhelp
--- nethack-3.4.0.orig/dat/wizhelp	Thu Apr 11 17:58:27 2002
+++ nethack-3.4.0/dat/wizhelp	Thu Apr 11 23:56:44 2002
@@ -15,6 +15,7 @@
 #monpoly_control == control monster polymorphs
 #poly == polymorph self
 #seenv == show seen vectors
+#showkills == show numbers of monsters killed
 #stats == show memory statistics
 #timeout == look at timeout queue
 #vision == show vision array
diff -ruN nethack-3.4.0.orig/doc/Guidebook.mn nethack-3.4.0/doc/Guidebook.mn
--- nethack-3.4.0.orig/doc/Guidebook.mn	Thu Apr 11 17:58:27 2002
+++ nethack-3.4.0/doc/Guidebook.mn	Thu Apr 11 23:56:44 2002
@@ -1971,6 +1971,10 @@
 Control what parts of the score list you are shown at the end (ex.
 ``scores:5 top scores/4 around my score/own scores'').  Only the first
 letter of each category (`t', `a', or `o') is necessary.
+.lp showborn
+When the game ends, show the number of each monster created
+in the "Vanquished creatures" list, if it differs from the
+number of those monsters killed.
 .lp showexp
 Show your accumulated experience points on bottom line (default off).
 .lp showscore
diff -ruN nethack-3.4.0.orig/doc/Guidebook.tex nethack-3.4.0/doc/Guidebook.tex
--- nethack-3.4.0.orig/doc/Guidebook.tex	Thu Apr 11 17:58:27 2002
+++ nethack-3.4.0/doc/Guidebook.tex	Thu Apr 11 23:56:44 2002
@@ -2421,6 +2421,11 @@
 ``{\tt scores:5top scores/4around my score/own scores}'').  Only the first
 letter of each category (`{\tt t}', `{\tt a}' or `{\tt o}') is necessary.
 %.lp
+\item[\ib{showborn}]
+When the game ends, show the number of each monster created
+in the ``Vanquished creatures'' list, if it differs from the
+number of those monsters killed.
+%.lp
 \item[\ib{showexp}]
 Show your accumulated experience points on bottom line (default off).
 %.lp
diff -ruN nethack-3.4.0.orig/doc/Guidebook.txt nethack-3.4.0/doc/Guidebook.txt
--- nethack-3.4.0.orig/doc/Guidebook.txt	Thu Apr 11 17:58:27 2002
+++ nethack-3.4.0/doc/Guidebook.txt	Thu Apr 11 23:56:44 2002
@@ -2590,6 +2590,11 @@
                scores'').   Only  the  first  letter of each category (`t',
                `a', or `o') is necessary.
 
+          showborn
+               When the game ends, show the number of each monster created
+               in the "Vanquished creatures" list, if it differs from the
+               number of those monsters killed.
+
           showexp
                Show your accumulated experience points on bottom line  (de-
                fault off).
diff -ruN nethack-3.4.0.orig/include/config.h nethack-3.4.0/include/config.h
--- nethack-3.4.0.orig/include/config.h	Thu Apr 11 19:50:23 2002
+++ nethack-3.4.0/include/config.h	Thu Apr 11 23:57:52 2002
@@ -348,6 +348,7 @@
  * bugs left here.
  */
 
+#define SHOW_BORN	/* show born patch */
 /*#define GOLDOBJ */	/* Gold is kept on obj chains - Helge Hafting */
 
 /* End of Section 5 */
diff -ruN nethack-3.4.0.orig/include/flag.h nethack-3.4.0/include/flag.h
--- nethack-3.4.0.orig/include/flag.h	Thu Apr 11 17:58:27 2002
+++ nethack-3.4.0/include/flag.h	Thu Apr 11 23:56:44 2002
@@ -208,6 +208,9 @@
 	boolean lan_mail;	/* mail is initialized */
 	boolean lan_mail_fetched; /* mail is awaiting display */
 #endif
+#ifdef SHOW_BORN
+	boolean show_born;	/* show numbers of created monsters */
+#endif
 /*
  * Window capability support.
  */
diff -ruN nethack-3.4.0.orig/src/cmd.c nethack-3.4.0/src/cmd.c
--- nethack-3.4.0.orig/src/cmd.c	Thu Apr 11 17:58:27 2002
+++ nethack-3.4.0/src/cmd.c	Thu Apr 11 23:56:44 2002
@@ -122,6 +122,8 @@
 STATIC_PTR int NDECL(wiz_show_vision);
 STATIC_PTR int NDECL(wiz_mon_polycontrol);
 STATIC_PTR int NDECL(wiz_show_wmodes);
+STATIC_PTR int NDECL(wiz_showkills);	/* showborn patch */
+extern void FDECL(list_vanquished, (int, BOOLEAN_P));	/* showborn patch */
 #if defined(__BORLANDC__) && !defined(_WIN32)
 extern void FDECL(show_borlandc_stats, (winid));
 #endif
@@ -719,6 +721,12 @@
 	return 0;
 }
 
+STATIC_PTR int wiz_showkills()	/* showborn patch */
+{
+	list_vanquished('y',FALSE);
+	return 0;
+}
+
 #endif /* WIZARD */
 
 
@@ -1409,6 +1417,7 @@
 	{(char *)0, (char *)0, donull, TRUE},
 	{(char *)0, (char *)0, donull, TRUE},
 	{(char *)0, (char *)0, donull, TRUE},
+	{(char *)0, (char *)0, donull, TRUE},	/* showborn patch */
         {(char *)0, (char *)0, donull, TRUE},
 	{(char *)0, (char *)0, donull, TRUE},
 	{(char *)0, (char *)0, donull, TRUE},
@@ -1427,6 +1436,7 @@
 	{"monpoly_control", "control monster polymorphs", wiz_mon_polycontrol, TRUE},
 	{"poly", "polymorph self", wiz_polyself, TRUE},
 	{"seenv", "show seen vectors", wiz_show_seenv, TRUE},
+	{"showkills", "show list of monsters killed", wiz_showkills, TRUE},
 	{"stats", "show memory statistics", wiz_show_stats, TRUE},
 	{"timeout", "look at timeout queue", wiz_timeout_queue, TRUE},
 	{"vision", "show vision array", wiz_show_vision, TRUE},
diff -ruN nethack-3.4.0.orig/src/end.c nethack-3.4.0/src/end.c
--- nethack-3.4.0.orig/src/end.c	Thu Apr 11 17:58:28 2002
+++ nethack-3.4.0/src/end.c	Thu Apr 11 23:56:44 2002
@@ -40,7 +40,7 @@
 STATIC_DCL void FDECL(add_artifact_score, (struct obj *));
 STATIC_DCL void FDECL(display_artifact_score, (struct obj *,winid));
 STATIC_DCL void FDECL(savelife, (int));
-STATIC_DCL void FDECL(list_vanquished, (int, BOOLEAN_P));
+void FDECL(list_vanquished, (int, BOOLEAN_P));	/* showborn patch */
 STATIC_DCL void FDECL(list_genocided, (int, BOOLEAN_P));
 STATIC_DCL boolean FDECL(should_query_disclose_option, (int, int*));
 
@@ -940,7 +940,7 @@
 	nethack_exit(status);
 }
 
-STATIC_OVL void
+void				/* showborn patch */
 list_vanquished(defquery, ask)
 int defquery;
 boolean ask;
@@ -997,6 +997,11 @@
 			else
 			    Sprintf(buf, "%d %s",
 				    nkilled, makeplural(mons[i].mname));
+#ifdef SHOW_BORN
+			if (iflags.show_born && nkilled != mvitals[i].born)
+			    Sprintf(buf + strlen(buf), " (%d created)",
+				   (int) mvitals[i].born);
+#endif
 		    }
 		    putstr(klwin, 0, buf);
 		}
diff -ruN nethack-3.4.0.orig/src/options.c nethack-3.4.0/src/options.c
--- nethack-3.4.0.orig/src/options.c	Thu Apr 11 17:58:28 2002
+++ nethack-3.4.0/src/options.c	Thu Apr 11 23:56:44 2002
@@ -156,6 +156,11 @@
 #else
 	{"sanity_check", (boolean *)0, FALSE, SET_IN_FILE},
 #endif
+#ifdef SHOW_BORN
+	{"showborn", &iflags.show_born, FALSE, SET_IN_GAME},
+#else
+	{"showborn", (boolean *)0, FALSE, SET_IN_FILE},
+#endif
 #ifdef EXP_ON_BOTL
 	{"showexp", &flags.showexp, FALSE, SET_IN_GAME},
 #else
