diff -ruN nethack-3.4.0.orig/src/hack.c nethack-3.4.0/src/hack.c
--- nethack-3.4.0.orig/src/hack.c	Thu Apr 11 17:58:28 2002
+++ nethack-3.4.0/src/hack.c	Fri Apr 12 00:09:08 2002
@@ -113,9 +113,14 @@
 		    (!mtmp->mtrapped ||
 			 !(ttmp && ((ttmp->ttyp == PIT) ||
 				    (ttmp->ttyp == SPIKED_PIT))))) {
-		if (canspotmon(mtmp))
-		    pline("There's %s on the other side.", mon_nam(mtmp));
-		else {
+		if (canspotmon(mtmp)) {
+		    boolean by_name = (mtmp->data->geno & G_UNIQ ||
+				       mtmp->isshk || mtmp->mnamelth);
+		    if (by_name && !Hallucination)
+			pline("%s is on the other side.", Monnam(mtmp));
+		    else
+			pline("There's %s on the other side.", a_monnam(mtmp));
+		} else {
 		    if (Blind) feel_location(sx, sy);
 		    You_hear("a monster behind %s.", the(xname(otmp)));
 		    map_invisible(rx, ry);
