diff -ruN nethack-3.4.0./util/makedefs.c nethack-3.4.0/util/makedefs.c
--- nethack-3.4.0./util/makedefs.c	Thu Apr 11 17:58:28 2002
+++ nethack-3.4.0/util/makedefs.c	Tue Jun 18 23:02:51 2002
@@ -592,12 +592,18 @@
 }
 
 static const char *build_opts[] = {
+#ifdef FLIPCOIN
+		"ability to flip coin",
+#endif
+		"additional levels",
 #ifdef AMIGA_WBENCH
 		"Amiga WorkBench support",
 #endif
 #ifdef ANSI_DEFAULT
 		"ANSI default terminal",
 #endif
+		"behind boulder patch",
+		"chivalry/bushido challenge",
 #ifdef TEXTCOLOR
 		"color",
 #endif
@@ -613,6 +619,10 @@
 #ifdef WIZARD
 		"debug mode",
 #endif
+		"dragon hoard",
+#ifdef DUMP_LOG
+		"dumping character to file",
+#endif
 #ifdef ELBERETH
 		"Elbereth",
 #endif
@@ -622,21 +632,30 @@
 #ifdef MFLOPPY
 		"floppy drive support",
 #endif
+#ifdef DUNGEON_GROWTH
+		"growth in dungeon",
+#endif
 #ifdef GOLDOBJ
 		"gold object in inventories",
 #endif
 #ifdef INSURANCE
 		"insurance files for recovering from crashes",
 #endif
+		"Kenny patch",
+#ifdef STEED
+		"key for riding",
+#endif
 #ifdef KOPS
 		"Keystone Kops",
 #endif
+		"listing seen monsters",
 #ifdef LOGFILE
 		"log file",
 #endif
 #ifdef MAIL
 		"mail daemon",
 #endif
+		"#monster messages",
 #ifdef GNUDOS
 		"MSDOS protected mode",
 #endif
@@ -694,9 +713,14 @@
 #ifdef SHELL
 		"shell command",
 #endif
+		"showing number of created monsters",
 #ifdef SINKS
 		"sinks",
 #endif
+#ifdef STICKY_OBJECTS
+		"sticky objects",
+#endif
+		"steed fix",
 #ifdef SUSPEND
 		"suspend command",
 #endif
@@ -722,6 +746,7 @@
 #ifdef WALLIFIED_MAZE
 		"walled mazes",
 #endif
+		"washing hands",
 #ifdef ZEROCOMP
 		"zero-compressed save files",
 #endif
