diff -ruN nethack-3.4.0.orig/src/mon.c nethack-3.4.0/src/mon.c
--- nethack-3.4.0.orig/src/mon.c	Thu Apr 11 17:58:28 2002
+++ nethack-3.4.0/src/mon.c	Sat Apr 13 16:46:02 2002
@@ -1578,6 +1578,7 @@
 int how;
 {
 	boolean be_sad = FALSE;		/* true if unseen pet is killed */
+	boolean kenny = (!strcmp(m_monnam(mdef), "Kenny"));
 
 	if ((mdef->wormno ? worm_known(mdef) : cansee(mdef->mx, mdef->my))
 		&& fltxt)
@@ -1595,8 +1596,15 @@
 	else
 	    mondied(mdef);
 
-	if (be_sad && mdef->mhp <= 0)
-	    You("have a sad feeling for a moment, then it passes.");
+	if (be_sad && mdef->mhp <= 0) {
+	    if (kenny || (Hallucination && !rn2(2))) {
+		verbalize("Oh my god, they killed Kenny!");
+		verbalize("You bastards!");
+	    } else {
+		You("have a %s feeling for a moment, then it passes.",
+		    (Hallucination ? "plaid" : "sad"));
+	    }
+	}
 }
 
 void
