diff -ruN nethack-3.4.0d/src/eat.c nethack-3.4.0/src/eat.c
--- nethack-3.4.0d/src/eat.c	Sat Aug 24 21:40:29 2002
+++ nethack-3.4.0/src/eat.c	Sat Aug 24 22:04:39 2002
@@ -777,6 +777,19 @@
 	if (eatmbuf) (void)eatmdone();
 
 	switch(pm) {
+	    case PM_NEWT:
+	      /* MRKR: "eye of newt" gives small magical energy boost */
+	      /* From an idea disussed in r.g.r.n by Rodney, Dayv and Rast*/
+	      
+	      You_feel("a mild buzz.");
+	      u.uen += rnd(3);
+	      if (u.uen > u.uenmax) {
+		u.uenmax++;
+		u.uen = u.uenmax;
+	      }
+	      flags.botl = 1;
+	      break;
+
 	    case PM_WRAITH:
 		pluslvl(FALSE);
 		break;
