diff -ruN nethack-3.4.1./include/config.h nethack-3.4.1/include/config.h
--- nethack-3.4.1./include/config.h	Fri Apr 18 20:52:27 2003
+++ nethack-3.4.1/include/config.h	Fri Apr 18 20:53:24 2003
@@ -44,7 +44,7 @@
  */
 #define TTY_GRAPHICS	/* good old tty based graphics */
 /* #define X11_GRAPHICS */	/* X11 interface */
-/* #define QT_GRAPHICS */	/* Qt interface */
+#define QT_GRAPHICS		/* Qt interface */
 /* #define GNOME_GRAPHICS */	/* Gnome interface */
 /* #define MSWIN_GRAPHICS */	/* Windows NT, CE, Graphics */
 
diff -ruN nethack-3.4.1./sys/unix/Makefile.src nethack-3.4.1/sys/unix/Makefile.src
--- nethack-3.4.1./sys/unix/Makefile.src	Fri Apr 18 20:52:27 2003
+++ nethack-3.4.1/sys/unix/Makefile.src	Fri Apr 18 20:53:45 2003
@@ -204,8 +204,8 @@
 
 #
 #
-WINSRC = $(WINTTYSRC)
-WINOBJ = $(WINTTYOBJ)
+WINSRC = $(WINTTYSRC) $(WINQTSRC)
+WINOBJ = $(WINTTYOBJ) $(WINQTOBJ)
 
 # on some systems the termcap library is in -ltermcap or -lcurses
 # on 386 Xenix, the -ltermlib tputs() seems not to work; use -lcurses instead
@@ -231,7 +231,7 @@
 # WINX11LIB = -lXaw -lXmu -lXpm -lXext -lXt -lX11 -lSM -lICE -lm # BSD/OS 2.0
 #
 # libraries for Qt
-WINQTLIB = -L$(QTDIR)/lib -lqt
+WINQTLIB = -L$(QTDIR)/lib -lqt-mt
 #
 # libraries for KDE (with Qt)
 WINKDELIB = -lkdecore -lkdeui -lXext
@@ -245,7 +245,7 @@
 # libraries for BeOS 
 WINBELIB = -lbe
 
-WINLIB = $(WINTTYLIB)
+WINLIB = $(WINTTYLIB) $(WINQTLIB)
 
 # any other strange libraries your system needs (for Sysunix only -- the more
 # specialized targets should already be right)
